---
title: 'PGC EWAS: Civilian Sensitivity Analyses'
date: "April 14, 2017"
output:
  pdf_document:
    fig_height: 6
    fig_width: 8
    toc: yes
    toc_depth: 3
  word_document:
    toc: yes
    toc_depth: '3'
bibliography: DNHS-GTP_NRG1.bibtex
---

```{r setup, include=FALSE}
rm(list=ls())
library(knitr)
library(ggplot2)
library(limma)
library(rmeta)
library(ChAMP)
library(forestplot)
library(qqman)
setwd("/Users/aratanat/Dropbox/Desktop_files/PGC_EWAS/Papers/NRG1/NRG1_Paper/")
```

# Summary

Two sets of sensitivity analyses were run to test their effect on the PGC EWAS Civilian NRG1 and HGS hits. First, two sets of meta-analyses were run: 1) including a dichotomous childhood maltreatment variable (1 = two or more types of childhood abuse) and 2) including an interaction term. Summary statistics (effect sizes, p-values, and FDR-adjusted p-values) were calculated for the PTSD, CM, and PTSDxCM parameters. In the model including CM, the effect of PTSD remained consistent and nominally significant for both CpG sites in addition to being FDR-significant for NRG1. The main effect of PTSD remained nominally significant for both sites when the PTSDxCM interction term was included. In neither model were CM or the PTSDxCM interaction term nominally significant. The second set of sensitivity analyses included both CM and current smoking as covariates and testing model fit in nested models. Results show that including current smoking was not a significant improvement in fit for either CpG site Overall, the main effects of PTSD on cg23637605 and cg19577098 were robust to inclusion of CM and smoking as covariates.

\newpage

# Methods

## Data

Data for this analysis came from DNHS and GTP PGC EWAS cohorts. For DNHS, the 100 individuals with either a past month diagnosis of PTSD (N = 40) or a lifetime history of no PTSD (N = 60) were included. There are 93 participants with a measure of childhood adversity, 92 of which also provided information on current smoking habits. For GTP, 265 participants were included, of which 74 had current PTSD and 191 were lifetime controls. All 265 individuals had smoking data.The 450K DNAm data was processed through the PGC EWAS pipeline and converted into m-values for analysis. 

### Childhood Maltreatment

Childhood event questions were drawn from the Conflict Tactics Scale (CTS) [@straus1979measuring], the Childhood Trauma Scale (CTQ) [@bernstein1997validity], and Wyatt’s eight-item interview guide as implemented by the Nurse’s Health Study II [@wyatt1985sexual]. CTQ items assessed physical abuse (e.g. “People in my family hit me so hard that it left me with bruises and marks”) and emotional abuse (e.g. “People in my family said hurtful or insulting things to me”) assessed before age 11. Response options were rated on a 5-point scale and ranged from “never true” (1) to “very often true” (5). CTS items assessed physical abuse before age 18 (e.g. “Did your parent, stepparent, or adult guardian ever push, grab, or shove you”) with response options ranging from “never” to “more than a few times.” Sexual abuse before age 18 was assessed with two items regarding sexual assault and rape (e.g. “Were you ever touched in a sexual way by an adult or older child”

In DNHS, the variable "CA_ordinary" can take on three values: 0 = no physical, sexual, or emotional maltreatment, 1 = one type of maltreatment, and 2 = two or more types. This was recoded into a dichotomous variable where 0 = no or one type of CM and 1 = two or more types of trauma. 

```{r, echo=F}
rm(list=ls())
dnhs.p<-read.csv("/Users/ly2207/Documents/Andrew/R/DNHS/DNHS_Methylation/phenotype/DNHS_methylation_pheno_111914_PTSDpm_Trauma_Check.csv", row.names=1, stringsAsFactors=F)

rownames(dnhs.p)<-dnhs.p$SentrixBarcode_Position

cases<-rownames(dnhs.p[dnhs.p$PTSDpm==1, ])
controls<-rownames(dnhs.p[dnhs.p$PTSDlife==0, ])
dnhs.p<-dnhs.p[append(cases, controls), ]
dnhs.p[is.na(dnhs.p$CA_ordinary), "CA_ordinary"]<-(99)

tab<-matrix(nrow=2, ncol=3)
colnames(tab)<-c("CM+", "CM-", "Missing")
rownames(tab)<-c("Case", "Control")

dnhs.p[dnhs.p$CA_ordinary<2, "CA_ordinary"]<-0
dnhs.p[dnhs.p$CA_ordinary==2, "CA_ordinary"]<-1
dnhs.p[dnhs.p$CA_ordinary==99, "CA_ordinary"]<-(-9)

tab["Case", "CM+"]<-nrow(dnhs.p[dnhs.p$CA_ordinary==1 & dnhs.p$PTSDpm==1, ])
tab["Case", "CM-"]<-nrow(dnhs.p[dnhs.p$CA_ordinary==0 & dnhs.p$PTSDpm==1, ])
tab["Case", "Missing"]<-nrow(dnhs.p[dnhs.p$CA_ordinary==-9 & dnhs.p$PTSDpm==1, ])

tab["Control", "CM+"]<-nrow(dnhs.p[dnhs.p$CA_ordinary==1 & dnhs.p$PTSDpm==0, ])
tab["Control", "CM-"]<-nrow(dnhs.p[dnhs.p$CA_ordinary==0 & dnhs.p$PTSDpm==0, ])
tab["Control", "Missing"]<-nrow(dnhs.p[dnhs.p$CA_ordinary==-9 & dnhs.p$PTSDpm==0, ])

Total<-apply(tab, 1, sum)
tab<-cbind(tab, Total)
Total<-apply(tab, 2, sum)
tab<-rbind(tab, Total)
dnhs<-tab
kable(tab, caption = "DNHS CM by PTSD status")
```

In GTP, Childhood event questions were drawn from the Childhood Trauma Scale (CTQ) [@bernstein1997validity]. Items for each type of abuse added together and transformed into categories of "None to Minimal", "Low to Moder", "Moderate to Severe", and "Severe to Extreme". The number of physical, sexual, and emotional abuse types categoriezed as "Moderate to Severe" were added together. Specifically, the variable "CHILDABphyssexemotandphysemotneg_ctq012_modandsev" takes on three values: 0 = no physical, sexual, or emotional maltreatment, 1 = one type of maltreatment, and 2 = two or more types. This will be recoded into a dichotomous variable where 0 = no or one type of CM and 1 = two or more types. 

```{r, echo=F}
rm(tab, cases, controls, Total)

# Adding CA into combined phenotype data
p1<-read.csv("/Users/ly2207/Documents/Andrew/R/GTP/Raw/GTPMetaAnalysisSamples_02.03.17.csv", 
             row.names=1, stringsAsFactors=F)

p2<-read.csv("/Users/ly2207/Documents/Andrew/R/GTP/Raw/DNHS_GTP_Phenotype_for_Analysis_02.26.17.csv", 
             row.names=1, stringsAsFactors=F)

rownames(p2)<-as.character(p2$ID)
int<-intersect(rownames(p1), rownames(p2))
p1<-p1[int, ]
p2<-p2[int, ]
p2$CA_ordinary<-p1$CHILDABphyssexemotandphysemotneg_ctq012_modandsev
gtp.p<-p2[!is.na(p2$PC1), ]

# Table of CA by PTSD in N = 265 analysis sample
gtp.p[is.na(gtp.p$CA_ordinary), "CA_ordinary"]<-99
gtp.p[gtp.p$CA_ordinary<2, "CA_ordinary"]<-0
gtp.p[gtp.p$CA_ordinary==2, "CA_ordinary"]<-1
gtp.p[gtp.p$CA_ordinary==99, "CA_ordinary"]<-(-9)

tab<-matrix(nrow=2, ncol=3)
colnames(tab)<-c("CM+", "CM-", "Missing")
rownames(tab)<-c("Case", "Control")

tab["Case", "CM+"]<-nrow(gtp.p[gtp.p$CA_ordinary==1 & gtp.p$PTSDcurr==1, ])
tab["Case", "CM-"]<-nrow(gtp.p[gtp.p$CA_ordinary==0 & gtp.p$PTSDcurr==1, ])
tab["Case", "Missing"]<-nrow(gtp.p[gtp.p$CA_ordinary==-9 & gtp.p$PTSDcurr==1, ])

tab["Control", "CM+"]<-nrow(gtp.p[gtp.p$CA_ordinary==1 & gtp.p$PTSDcurr==0, ])
tab["Control", "CM-"]<-nrow(gtp.p[gtp.p$CA_ordinary==0 & gtp.p$PTSDcurr==0, ])
tab["Control", "Missing"]<-nrow(gtp.p[gtp.p$CA_ordinary==-9 & gtp.p$PTSDcurr==0, ])

Total<-apply(tab, 1, sum)
tab<-cbind(tab, Total)
Total<-apply(tab, 2, sum)
tab<-rbind(tab, Total)
gtp<-tab
kable(tab, caption="GTP CM by PTSD status")
```

In the combined sample, there are a total of 114 cases and 251 controls with 13 individuals having missing childhood maltreatment status:

```{r, echo=F}
rm(tab, Total, p2, p1, int)
tab<-dnhs+gtp
kable(tab, caption="Combined sample CM by PTSD status")
```

In the final analytical sample, the dsitribution of childhood maltreatment across PTSD case status differs significantly, as we would expect:

```{r, echo=F}
rm(tab)
tab<-dnhs+gtp
tab<-tab[1:2, 1:2]
Total<-apply(tab, 1, sum)
tab<-cbind(tab, Total)
Total<-apply(tab, 2, sum)
tab<-rbind(tab, Total)
Chi.square<-matrix(c("", "X-squared = ", round(chisq.test(tab[1:2, 1:2])$statistic, 2),
                     "", "p-value", signif(chisq.test(tab[1:2, 1:2])$p.value,3)), nrow=2, byrow=T)

tab<-rbind(tab, Chi.square)
kable(tab, caption="Combined sample CM by PTSD status, missings removed")
```

### Current Smoking

In DNHS, current smoking was defined by asking participants how many cigarettes they had smoked in the past month. Those who had smoked one or more were considered current smokers. 

```{r, echo=F}
rm(tab, Total, gtp, dnhs, Chi.square)
p1<-read.csv("/Users/ly2207/Documents/Andrew/R/DNHS/DNHS_Methylation/phenotype/DNHS179_meta_pheno_smoking_11.07.16.csv",
             row.names=1, stringsAsFactors=F)
rownames(p1)<-p1$SentrixBarcode_Position

#all(rownames(dnhs.p)%in%rownames(p1))
p1<-p1[rownames(dnhs.p), ]
#all(rownames(p1)==rownames(dnhs.p))
dnhs.p$CurrSmoke<-p1$CurrSmoke

tab<-matrix(nrow=2, ncol=3)
colnames(tab)<-c("Smoker+", "Smoker-", "Missing")
rownames(tab)<-c("Case", "Control")

tab["Case", "Smoker+"]<-length(which(dnhs.p$PTSDpm==1 & dnhs.p$CurrSmoke==1))
tab["Case", "Smoker-"]<-length(which(dnhs.p$PTSDpm==1 & dnhs.p$CurrSmoke==0))
tab["Case", "Missing"]<-length(which(dnhs.p$PTSDpm==1 & is.na(dnhs.p$CurrSmoke)))

tab["Control", "Smoker+"]<-length(which(dnhs.p$PTSDpm==0 & dnhs.p$CurrSmoke==1))
tab["Control", "Smoker-"]<-length(which(dnhs.p$PTSDpm==0 & dnhs.p$CurrSmoke==0))
tab["Control", "Missing"]<-length(which(dnhs.p$PTSDpm==0 & is.na(dnhs.p$CurrSmoke)))

Total<-apply(tab, 1, sum)
tab<-cbind(tab, Total)
Total<-apply(tab, 2, sum)
tab<-rbind(tab, Total)
dnhs<-tab
kable(tab, caption="DNHS current smoking status by PTSD status")

```

In GTP, [Andrew: I actually don’t know how current smoking was determined].

```{r, echo=F}
rm(p1, tab, Total)

tab<-matrix(nrow=2, ncol=3)
colnames(tab)<-c("Smoker+", "Smoker-", "Missing")
rownames(tab)<-c("Case", "Control")

tab["Case", "Smoker+"]<-length(which(gtp.p$PTSDcurr==1 & gtp.p$SmokerCurr==1))
tab["Case", "Smoker-"]<-length(which(gtp.p$PTSDcurr==1 & gtp.p$SmokerCurr==0))
tab["Case", "Missing"]<-length(which(gtp.p$PTSDcurr==1 & is.na(gtp.p$SmokerCurr)))

tab["Control", "Smoker+"]<-length(which(gtp.p$PTSDcurr==0 & gtp.p$SmokerCurr==1))
tab["Control", "Smoker-"]<-length(which(gtp.p$PTSDcurr==0 & gtp.p$SmokerCurr==0))
tab["Control", "Missing"]<-length(which(gtp.p$PTSDcurr==0 & is.na(gtp.p$SmokerCurr)))

Total<-apply(tab, 1, sum)
tab<-cbind(tab, Total)
Total<-apply(tab, 2, sum)
tab<-rbind(tab, Total)
gtp<-tab
kable(tab, caption="GTP current smoking status by PTSD status")
```

Combined Sample:

```{r, echo=F}
rm(tab, Total)
tab<-dnhs+gtp
kable(tab, caption="Combined sample CM by PTSD status")
```

With the one missing individual from DNHS dropped, as expected smoking status differs by PTSD status: 

```{r, echo=F}
rm(tab)
tab<-dnhs+gtp
tab<-tab[1:2, 1:2]
Total<-apply(tab, 1, sum)
tab<-cbind(tab, Total)
Total<-apply(tab, 2, sum)
tab<-rbind(tab, Total)
Chi.square<-matrix(c("", "X-squared = ", round(chisq.test(tab[1:2, 1:2])$statistic, 2),
                     "", "p-value", signif(chisq.test(tab[1:2, 1:2])$p.value,3)), nrow=2, byrow=T)

tab<-rbind(tab, Chi.square)
kable(tab, caption="Combined sample CM by PTSD status, missings removed")
```

## Statistical Models: 

### Childhood Maltreatment EWAS Models:

Within each cohort the following models were fit:

Model 1: $M_{CpG} = \beta*X + \gamma_1*PTSD + \gamma_2*CM$

Model 2: $M_{CpG} = \beta*X + \gamma_1*PTSD + \gamma_2*CM + \gamma_3*CM*PTSD$

where X is a matrix of covariates including age, gender, cell types, and population-stratification PCs. Regarding PCs, for GTP, these included PCs 1, 2 and 3 from GWAS data and for DNHS they included PCs 2, 3, and 4 from EWAS data using the Barfield method per the PGC EWAS pipeline. These models were fit using the R package limma and empirical Bayes standard errors calculated. The Benjamini-Hochberg False Discovery Rate method was used to adjust for multiple hypothesis testing.

These results were then meta-analyzed using the PGC EWAS inverse normal pipeline to obtain a p-value and FDR-adjusted p-value for each CpG site for the effect of PTSD, CM, and the interaction term CMxPTSD across each cohort. The FDR-adjusted p-value was only for the number of tests of that coefficient, not the overall number of tests in each model or across analyses (e.g., 450k hypotheses / CpG sites). To obtain a point estimate and confidence interval of effect, $\beta$ coefficients for each cohorts were combined using generic inverse variance meta-analysis where the estimated effect ($\hat{\theta}$) was calculated as:

$\hat{\theta} = \frac{\sum_{1}^{k}w_i\hat{\theta}_i}{\sum_{1}^{k}w_i}$

where 

$w_i = \frac{1}{\hat{\sigma}^2_{\theta_i}}$

$var(\hat{\theta}) = \frac{1}{\sum_{1}^{k}w_i}$

95% CI: $\hat{\theta}-1.96\sqrt{var(\hat{\theta})}$ to  $\hat{\theta}+1.96\sqrt{var(\hat{\theta})}$

The point estimate and confidence interval from the generic inverse variance method were used to create the forest plots.

### Smoking ANOVAs

Per Don Armstrong's suggestion, three OLS models without the empirical Bayes adjustment were run only for the CpG sites cg23637605 (*NRG1*) and cg19577098 (*HGS*):

1. Full Cohort (= "FULL"): Model 1 and Model 2 were run in the both cohorts with all participants.

2. Smoking Cohort (= "SMOKE-"), but current smoking not included in the model: Model 1 and Model 2 were run in the both cohorts with only the participants with smoking data.

3. Smoking Cohort (="SMOKE+") plus current smoking included in the model: The models were updated to include currenting smoking. Thus:

Model 1: $M_{CpG} = \beta*X + \gamma_1*PTSD + \gamma_2*CM + \gamma_3*Smoking$

Model 2: $M_{CpG} = \beta*X + \gamma_1*PTSD + \gamma_2*CM + \gamma_3*CM*PTSD + \gamma_4*Smoking$

The results from the FULL cohort were compared with teh SMOKE- cohort to see if there were any significant differences due to decreased sample size. After that the SMOKE- cohort was compared with the SMOKE+ chort to see if the inclusion of current smoking modified the PTSD and CM results. In addition, the fits of the SMOKE- and SMOKE+ models were compared using AIC.

**Note: originally there were significantly different number of individuals with and without current smoking data. Since November, we fixed DNHS and I received updated phenotypes from Alicia Smith as well. While there are not many differences between Models 1 and 2, I present both here for historical reasons.**

\newpage

# Results

## Childhood Maltreatment

In the main effects model (Model 1, no PTSDxCM interaction), childhood maltreatment was nominally insignificant in the DNHS, GTP, and Meta-Analyses for both *NRG1* and *HGS* CpG sites.  In the interaction model, both childhood maltreatment and the interaction term were insignificant. The main effect of PTSD was FDR-significant in Model 1 for *NRG1* with a consistent effect and nominally significant in Model 2, which may be due to the interaction term siphoning off power for the PTSD main effect. *HGS* was nominally significant in both Model 1 and 2.  

### Model 1: Main Effects

The tables below list the $\beta$ coefficients, standard errors, p-values, FDR-adjusted p-values, $\lambda$ for each PTSD/CM variable, and sample sizes from Model 1 for cg23637605 and cg19577098. For both CpG sites, the association between PTSD and DNAm is consistent between DNHS and GTP, with cases having lower DNAm than controls on average. For both CpG sites and in neither cohort nor in the meta-analysis is the effect of Childhood Maltreatment significant or in a consistent direction of effect.

**cg23637605:**

```{r, echo=F}
rm(list=ls())
tab<-read.csv("PGC_EWAS_NRG1_PTSD_mod1_cg23637605.csv", row.names=1)
kable(tab)
```

\newpage

!["Model 1: PTSD Coefficient"](PGC_EWAS_NRG1_PTSD_mod1_cg23637605.png)

\newpage

**cg19577098:**

```{r, echo=F}
rm(list=ls())
tab<-read.csv("PGC_EWAS_NRG1_PTSD_mod1_cg19577098.csv", row.names=1)
kable(tab, caption="Main Effects for cg19577098.")
```

!["Model 1: PTSD Coefficient"](PGC_EWAS_NRG1_PTSD_mod1_cg19577098.png)

\newpage

### Model 2: Interaction Effects

The tables below lists the $\beta$ coefficients, standard errors, p-values, FDR-adjusted p-values, $\lambda$ for each PTSD/CM variable, and sample sizes from Model 2. For both CpG sites, the main effect of PTSD remains consistent between Model 1 and Model 2 with participants with current PTSD having lower DNAm on average and neither CM nor the interaction term of CM and PTSD are significant in either cohort or in the meta-analysis. Neither of the main effects for PTSD are significant but again this may be due to the insignificant interaction term draining some power from the main effect.

**cg23637605:**

```{r, echo=F}
rm(list=ls())
tab<-read.csv("PGC_EWAS_NRG1_PTSD_mod2_cg23637605.csv", row.names=1)
kable(tab)
```

\newpage

!["Model 2: PTSD Coefficient"](PGC_EWAS_NRG1_PTSD_mod2_cg23637605.png)

\newpage

**cg19577098:**

```{r, echo=F}
rm(list=ls())
tab<-read.csv("PGC_EWAS_NRG1_PTSD_mod2_cg19577098.csv", row.names=1)
kable(tab)
```

!["Model 2: PTSD Coefficient"](PGC_EWAS_NRG1_PTSD_mod2_cg19577098.png)

\newpage

## Smoking Sensitivity Analysis

The full results for both models and CpG sites are below. For *NRG1*, in Model 1 (no interaction model) the $\beta$ coefficient for PTSD remained in the same direction and relatively same magnitude for both DNHS and GTP across all three model. The $\beta$ coefficients for CM were in opposite directions between the cohorts but remained stable across the models as well. In both cohorts, the SMOKE+ model was not a significant improvement compared to SMOKE- in terms of model fit with the ANOVA p-value greater than 0.05 (DNHS = 0.696; GTP = 0.775). Similarly for *HGS*, the beta coefficients for PTSD remained in the same direction although for DNHS the main effect was not significant (p= 0.054). In neither cohort did the including smoking improve the fit of the model nor was CM significant.

Results were similar for Model 2, the interaction model, with the SMOKE+ model not an improvement over the SMOKE- model. In DNHS, the inclusion of the PTSDxCM interaction term did lead to the main effect of PTSD becoming significant for *HGS*. In addition, all $\beta$ PTSD coefficients showed consistency of direction across all models while coefficient for smoking, CM, or the PTSDxCM interaction were not significant in any model for either cohort.

### Model 1:

#### DNHS:

**cg23637605:**

```{r, echo=F}
rm(list=ls())
dnhs<-read.csv("DNHS_mod1_smoking_results.csv", stringsAsFactors=F, row.names=1)
dnhs[which(is.na(dnhs), arr.ind=T)]<-""
colnames(dnhs)<-c("Main", "Main w/ Smoke N", "Main + Smoking")
kable(dnhs)
```

**cg19577098:**
```{r, echo=F}
rm(list=ls())
dnhs<-read.csv("DNHS_mod1_smoking_cg19577098_results.csv", stringsAsFactors=F, row.names=1)
dnhs[which(is.na(dnhs), arr.ind=T)]<-""
colnames(dnhs)<-c("Main", "Main w/ Smoke N", "Main + Smoking")
kable(dnhs)
```


#### GTP

**cg23637605:**

```{r, echo=F}
rm(list=ls())
gtp<-read.csv("GTP_mod1_smoking_results.csv", stringsAsFactors=F, row.names=1)
gtp[which(is.na(gtp), arr.ind=T)]<-""
colnames(gtp)<-c("Main", "Main w/ Smoke N", "Main + Smoking")
kable(gtp)
```

**cg19577098:**

```{r, echo=F}
rm(list=ls())
gtp<-read.csv("GTP_mod1_smoking_cg19577098_results.csv", stringsAsFactors=F, row.names=1)
gtp[which(is.na(gtp), arr.ind=T)]<-""
colnames(gtp)<-c("Main", "Main w/ Smoke N", "Main + Smoking")
kable(gtp)
```

### Model 2:

#### DNHS:

**cg23637605:**

```{r, echo=F}
rm(list=ls())
dnhs<-read.csv("DNHS_mod2_smoking_results.csv", stringsAsFactors=F, row.names=1)
dnhs[which(is.na(dnhs), arr.ind=T)]<-""
colnames(dnhs)<-c("Main", "Main w/ Smoke N", "Main + Smoking")
kable(dnhs)
```

**cg19577098:**

```{r, echo=F}
rm(list=ls())
dnhs<-read.csv("DNHS_mod2_smoking_cg19577098_results.csv", stringsAsFactors=F, row.names=1)
dnhs[which(is.na(dnhs), arr.ind=T)]<-""
colnames(dnhs)<-c("Main", "Main w/ Smoke N", "Main + Smoking")
kable(dnhs)
```

#### GTP

**cg23637605:**

```{r, echo=F}
rm(list=ls())
gtp<-read.csv("GTP_mod2_smoking_cg19577098_results.csv", stringsAsFactors=F, row.names=1)
gtp[which(is.na(gtp), arr.ind=T)]<-""
colnames(gtp)<-c("Main", "Main w/ Smoke N", "Main + Smoking")
kable(gtp)
```

**cg19577098:**

```{r, echo=F}
rm(list=ls())
gtp<-read.csv("GTP_mod2_smoking_cg19577098_results.csv", stringsAsFactors=F, row.names=1)
gtp[which(is.na(gtp), arr.ind=T)]<-""
colnames(gtp)<-c("Main", "Main w/ Smoke N", "Main + Smoking")
kable(gtp)
```

\newpage

# QQ Plots

## Model 1:

!["Model 1: PTSD Coefficient"](PGC_EWAS_NRG1_PTSD_mod1_QQplot.png)

!["Model 1: CA Coefficient"](PGC_EWAS_NRG1_CA_mod1_QQplot.png)

\newpage

## Model 2: 

!["Model 2: PTSD Coefficient"](PGC_EWAS_NRG1_PTSD_mod2_QQplot.png)

!["Model 2: CA Coefficient"](PGC_EWAS_NRG1_CA_mod2_QQplot.png)

!["Model 2: PTSDxCA Coefficient"](PGC_EWAS_NRG1_PTSDxCA_mod2_QQplot.png)

\newpage

# Code Appendix

## Individual Study EWAS Code:

### DNHS EWAS: Models 1 and 2

```{r, echo=T, eval=F}
dnhs.p<-dnhs.p[dnhs.p$CA_ordinary>=0,]

load("/Users/ly2207/Documents/Andrew/R/DNHS/DNHS_Methylation/GenomeStudio/DNHS192_bkgd_beta_postComBat_bySample.Rdata")
beta.norm<-reversbeta
rm(reversbeta)

pheno<-dnhs.p
all(rownames(pheno)%in%colnames(beta.norm))
beta.norm<-beta.norm[, rownames(pheno)]

dim(beta.norm)
all(rownames(pheno)==colnames(beta.norm))

# Assuming "beta.norm" is the normalized beta matrix, check the ranges of the beta values. 
# Beta values of 0 or 1 will be transformed into M-values of "-Inf". 
range(beta.norm, na.rm=T)

# Changing beta values of 0 to 0.0001
if(min(beta.norm, na.rm=T)==0){
  beta.norm[which(beta.norm==0)]<-0.0001
}

# Changing beta values of 1 to 0.9999
if(max(beta.norm, na.rm=T)==1){
  beta.norm[which(beta.norm==1)]<-0.9999
}

range(beta.norm, na.rm=T)
sum(is.na(beta.norm))

# Convert to Mvalues using log2
beta.norm<-log2(beta.norm/(1-beta.norm)) # log transforming
#sum(beta.norm=="-Inf", na.rm=T) # Should be 0
sum(is.na(beta.norm)) # should be same as the number of missing in the beta.norm matrix

# Check that the Subjects in the pheno rows matches order of subjects in Beta columns

all(rownames(pheno)==colnames(beta.norm)) 
sum(is.na(match(rownames(pheno), colnames(beta.norm)))) 
pheno<-pheno[colnames(beta.norm),]
all(rownames(pheno)==colnames(beta.norm)) # this should be true


########################################################################################
# Step 2: Run Model 1
########################################################################################

# Create Design Matrix: Age, Gender (if applicable), PCs from Pop strat, cell types, PTSD Current
design<-model.matrix(~age+female+Comp.2+Comp.3+Comp.4+CD8T+CD4T+NK+Bcell+Mono+PTSDpm+CA_ordinary, 
                     data=pheno)

fit<-lmFit(beta.norm, design) # Runs linear models
fit.coef<-fit$coef # Extracts the beta coefficients
all(rownames(fit.coef)==rownames(beta.norm)) # Should be TRUE

# Calculate the number of subjects with complete beta values
# This will be used for the weight in the meta-analysis
N.subjects<-apply(beta.norm, 1, function(x) sum(!is.na(x)))

fit.coef<-cbind(fit.coef, N.subjects) # Adds N subjects to the coefficients for use in weighting

contrast.matrix<-makeContrasts(PTSDpm, levels=design)
contrast.matrix # Contrast matrix extracts only the coefficient we are interested in
rownames(contrast.matrix)[1]<-"(Intercept)" # Have to rename the contrast matrix
contrast.matrix

fit2<-contrasts.fit(fit, contrast.matrix)
fit2.ebayes<-eBayes(fit2) # Run empirical bayes
results<-topTable(fit2.ebayes,number=nrow(fit2.ebayes),coef=1, adjust="BH")

contrast.matrix.ca<-makeContrasts(CA_ordinary, levels=design)
rownames(contrast.matrix.ca)[1]<-"(Intercept)" # Have to rename the contrast matrix
contrast.matrix.ca

fit2.ca<-contrasts.fit(fit, contrast.matrix.ca)
fit2.ca.ebayes<-eBayes(fit2.ca) # Run empirical bayes
results.ca<-topTable(fit2.ca.ebayes,number=nrow(fit2.ca.ebayes),coef=1, adjust="BH")

save(fit2.ebayes, fit.coef, results, fit2.ca.ebayes, fit2.ca, results.ca, 
     file="DNHS_NRG1_PTSD_CM_mod1_10.21.16.Rdata")

rm(fit, fit.coef, N.subjects, contrast.matrix, fit2, results, fit2.ebayes, design,
   fit2.ca, fit2.ca.ebayes, contrast.matrix.ca, results.ca)

########################################################################################
# Step 2: Run Model 2
########################################################################################

# Create Design Matrix: Age, Gender (if applicable), PCs from Pop strat, cell types, PTSD Current
design<-model.matrix(~age+female+Comp.2+Comp.3+Comp.4+CD8T+CD4T+NK+Bcell+Mono+PTSDpm+CA_ordinary +
                       PTSDpm*CA_ordinary, data=pheno)

colnames(design)<-gsub(":", "X", colnames(design))

fit<-lmFit(beta.norm, design) # Runs linear models
fit.coef<-fit$coef # Extracts the beta coefficients
all(rownames(fit.coef)==rownames(beta.norm)) # Should be TRUE

# Calculate the number of subjects with complete beta values
# This will be used for the weight in the meta-analysis
N.subjects<-apply(beta.norm, 1, function(x) sum(!is.na(x)))

fit.coef<-cbind(fit.coef, N.subjects) # Adds N subjects to the coefficients for use in weighting
contrast.matrix<-makeContrasts(PTSDpm, levels=design)
contrast.matrix # Contrast matrix extracts only the coefficient we are interested in
rownames(contrast.matrix)[1]<-"(Intercept)" # Have to rename the contrast matrix
contrast.matrix

fit2<-contrasts.fit(fit, contrast.matrix)
fit2.ebayes<-eBayes(fit2) # Run empirical bayes
results<-topTable(fit2.ebayes,number=nrow(fit2.ebayes),coef=1, adjust="BH")

contrast.matrix.ca<-makeContrasts(CA_ordinary, levels=design)
rownames(contrast.matrix.ca)[1]<-"(Intercept)" # Have to rename the contrast matrix
contrast.matrix.ca

fit2.ca<-contrasts.fit(fit, contrast.matrix.ca)
fit2.ca.ebayes<-eBayes(fit2.ca) # Run empirical bayes
results.ca<-topTable(fit2.ca.ebayes,number=nrow(fit2.ca.ebayes),coef=1, adjust="BH")

contrast.matrix.caPTSD<-makeContrasts(PTSDpmXCA_ordinary, levels=design)
rownames(contrast.matrix.caPTSD)[1]<-"(Intercept)" # Have to rename the contrast matrix
contrast.matrix.caPTSD

fit2.caPTSD<-contrasts.fit(fit, contrast.matrix.caPTSD)
fit2.caPTSD.ebayes<-eBayes(fit2.caPTSD) # Run empirical bayes
results.caPTSD<-topTable(fit2.caPTSD.ebayes,number=nrow(fit2.caPTSD.ebayes),coef=1, adjust="BH")

save(fit2.ebayes, fit.coef, results, fit2.ca.ebayes, fit2.ca, results.ca, 
    fit2.caPTSD.ebayes, fit2.caPTSD, results.caPTSD,
     file="DNHS_NRG1_PTSD_CM_mod2_10.21.16.Rdata")

rm(fit, fit.coef, N.subjects, contrast.matrix, fit2, results, fit2.ebayes, design, fit2.caPTSD,
   fit2.caPTSD.ebayes, results.caPTSD, contrast.matrix.caPTSD, fit2.ca, fit2.ca.ebayes, results.ca,
   contrast.matrix.ca)
rm(list=ls()[-match(c("gtp.p", "dnhs.p"), ls())])
```

\newpage

### GTP EWAS: Models 1 and 2

```{r, echo=T, eval=F}
gtp.p<-gtp.p[gtp.p$CA_ordinary>=0,]

load("/Users/ly2207/Documents/Andrew/R/GTP/Emory_QC/GTP283_BMIQ_postComBat.Rdata")

beta.norm<-as.matrix(beta)
str(beta.norm)
all(beta.norm==beta, na.rm=T)
rm(beta)

pheno<-gtp.p
all(rownames(pheno)%in%colnames(beta.norm))
beta.norm<-beta.norm[, rownames(pheno)]

dim(beta.norm)
all(rownames(pheno)==colnames(beta.norm))

# Assuming "beta.norm" is the normalized beta matrix, check the ranges of the beta values. 
# Beta values of 0 or 1 will be transformed into M-values of "-Inf". 
range(beta.norm, na.rm=T)

# Changing beta values of 0 to 0.0001
if(min(beta.norm, na.rm=T)==0){
  beta.norm[which(beta.norm==0)]<-0.0001
}

# Changing beta values of 1 to 0.9999
if(max(beta.norm, na.rm=T)==1){
  beta.norm[which(beta.norm==1)]<-0.9999
}

range(beta.norm, na.rm=T)
sum(is.na(beta.norm))

# Convert to Mvalues using log2
beta.norm<-log2(beta.norm/(1-beta.norm)) # log transforming
#sum(beta.norm=="-Inf", na.rm=T) # Should be 0
#sum(is.na(beta.norm)) # should be same as the number of missing in the beta.norm matrix

# Check that the Subjects in the pheno rows matches order of subjects in Beta columns
all(rownames(pheno)==colnames(beta.norm)) 
sum(is.na(match(rownames(pheno), colnames(beta.norm)))) 
pheno<-pheno[colnames(beta.norm),]
all(rownames(pheno)==colnames(beta.norm)) # this should be true

########################################################################################
# Step 2: Run Model 1
########################################################################################
pheno$Gender<-factor(pheno$Gender, levels=c("Male", "Female"))
colnames(pheno)
# Create Design Matrix: Age, Gender (if applicable), PCs from Pop strat, cell types, PTSD Current
design<-model.matrix(~Age+Gender+PC1+PC2+PC3+CD8T+CD4T+NK+Bcell+Mono+PTSDcurr+
                       CA_ordinary, data=pheno)

fit<-lmFit(beta.norm, design) # Runs linear models
fit.coef<-fit$coef # Extracts the beta coefficients
all(rownames(fit.coef)==rownames(beta.norm)) # Should be TRUE

# Calculate the number of subjects with complete beta values
# This will be used for the weight in the meta-analysis
N.subjects<-apply(beta.norm, 1, function(x) sum(!is.na(x)))

fit.coef<-cbind(fit.coef, N.subjects) # Adds N subjects to the coefficients for use in weighting

contrast.matrix<-makeContrasts(PTSDcurr, levels=design)
contrast.matrix # Contrast matrix extracts only the coefficient we are interested in
rownames(contrast.matrix)[1]<-"(Intercept)" # Have to rename the contrast matrix
contrast.matrix

fit2<-contrasts.fit(fit, contrast.matrix)
fit2.ebayes<-eBayes(fit2) # Run empirical bayes
results<-topTable(fit2.ebayes,number=nrow(fit2.ebayes),coef=1, adjust="BH")

contrast.matrix.ca<-makeContrasts(CA_ordinary, levels=design)
rownames(contrast.matrix.ca)[1]<-"(Intercept)" # Have to rename the contrast matrix
contrast.matrix.ca

fit2.ca<-contrasts.fit(fit, contrast.matrix.ca)
fit2.ca.ebayes<-eBayes(fit2.ca) # Run empirical bayes
results.ca<-topTable(fit2.ca.ebayes,number=nrow(fit2.ca.ebayes),coef=1, adjust="BH")

save(fit2.ebayes, fit.coef, results, fit2.ca.ebayes, fit2.ca, results.ca, 
     file="GTP_NRG1_PTSD_CM_mod1_03.08.17.Rdata")

rm(fit, fit.coef, N.subjects, contrast.matrix, fit2, results, fit2.ebayes, design,
   fit2.ca, fit2.ca.ebayes, results.ca, contrast.matrix.ca)

########################################################################################
# Step 2: Run Model 2
########################################################################################

# Create Design Matrix: Age, Gender (if applicable), PCs from Pop strat, cell types, PTSD Current
design<-model.matrix(~Age+Gender+PC1+PC2+PC3+CD8T+CD4T+NK+Bcell+Mono+PTSDcurr+
                       CA_ordinary+PTSDcurr*CA_ordinary, data=pheno)

colnames(design)<-gsub(":", "X", colnames(design))

all(rownames(design)==colnames(beta.norm))

fit<-lmFit(beta.norm, design) # Runs linear models
fit.coef<-fit$coef # Extracts the beta coefficients
all(rownames(fit.coef)==rownames(beta.norm)) # Should be TRUE

# Calculate the number of subjects with complete beta values
# This will be used for the weight in the meta-analysis
N.subjects<-apply(beta.norm, 1, function(x) sum(!is.na(x)))

fit.coef<-cbind(fit.coef, N.subjects) # Adds N subjects to the coefficients for use in weighting
contrast.matrix<-makeContrasts(PTSDcurr, levels=design)
contrast.matrix # Contrast matrix extracts only the coefficient we are interested in
rownames(contrast.matrix)[1]<-"(Intercept)" # Have to rename the contrast matrix
contrast.matrix

fit2<-contrasts.fit(fit, contrast.matrix)
fit2.ebayes<-eBayes(fit2) # Run empirical bayes
results<-topTable(fit2.ebayes,number=nrow(fit2.ebayes),coef=1, adjust="BH")

contrast.matrix.ca<-makeContrasts(CA_ordinary, levels=design)
rownames(contrast.matrix.ca)[1]<-"(Intercept)" # Have to rename the contrast matrix
contrast.matrix.ca

fit2.ca<-contrasts.fit(fit, contrast.matrix.ca)
fit2.ca.ebayes<-eBayes(fit2.ca) # Run empirical bayes
results.ca<-topTable(fit2.ca.ebayes,number=nrow(fit2.ca.ebayes),coef=1, adjust="BH")

contrast.matrix.caPTSD<-makeContrasts(PTSDcurrXCA_ordinary, levels=design)
rownames(contrast.matrix.caPTSD)[1]<-"(Intercept)" # Have to rename the contrast matrix
contrast.matrix.caPTSD

fit2.caPTSD<-contrasts.fit(fit, contrast.matrix.caPTSD)
fit2.caPTSD.ebayes<-eBayes(fit2.caPTSD) # Run empirical bayes
results.caPTSD<-topTable(fit2.caPTSD.ebayes,number=nrow(fit2.caPTSD.ebayes),coef=1, adjust="BH")

save(fit2.ebayes, fit.coef, results, fit2.ca.ebayes, fit2.ca, results.ca, 
    fit2.caPTSD.ebayes, fit2.caPTSD, results.caPTSD,
    file="GTP_NRG1_PTSD_CM_mod2_03.08.17.Rdata")

rm(list=ls())
```

\newpage

## Meta-Analysis Code

### Model 1:

#### PTSD Main Effect Meta-Analysis

```{r, eval=F, echo=T}

rm(list=ls())

########################################################################################
# Step 1: Load Summary Statistics 
########################################################################################

# Step 1A: Load DNHS Results
load("DNHS_NRG1_PTSD_CM_mod1_10.21.16.Rdata")
rm(results.ca, fit2.ca, fit2.ca.ebayes)
DNHS.coef<-fit.coef
rm(fit.coef)
results<-results[order(results[, "P.Value"]), ]
rank<-c(1:nrow(results))
results<-cbind(results, rank)
DNHS.results<-results
rm(results, rank)
DNHS.ebayes<-fit2.ebayes
rm(fit2.ebayes)

# Step 1B: Load GTP Results
load("GTP_NRG1_PTSD_CM_mod1_03.08.17.Rdata")
rm(results.ca, fit2.ca, fit2.ca.ebayes)
GTP.coef<-fit.coef
rm(fit.coef)
results<-results[order(results[, "P.Value"]), ]
GTP.results<-results
rm(results)
rank<-c(1:nrow(GTP.results))
GTP.results<-cbind(GTP.results, rank)
rm(rank)
GTP.ebayes<-fit2.ebayes
rm(fit2.ebayes)

save.image("PGC_EWAS_NRG1_PTSD_mod1.Rdata")
rm(list=ls())

########################################################################################
# # Step 2: Summary of Results
########################################################################################

load("PGC_EWAS_NRG1_PTSD_mod1.Rdata")

studies<-c("DNHS", "GTP")
tab<-matrix(nrow=length(studies), ncol=9)
colnames(tab)<-c("Study", "Sites", "Min N", "Max N", "Lambda", "FDR", "p<5x10^-5",
                 "p<5x10^-6", "p<5x10^-7")

for(ii in 1:length(studies)){
  tab[ii, "Study"]<-studies[ii]
  res<-get(paste(studies[ii], ".results", sep=""))
  res<-res[!is.na(res[, "logFC"]),] 
  tab[ii, "p<5x10^-5"]<-sum(res[, "P.Value"]<(5*10^(-5))) 
  tab[ii, "p<5x10^-6"]<-sum(res[, "P.Value"]<(5*10^(-6))) 
  tab[ii, "p<5x10^-7"]<-sum(res[, "P.Value"]<(5*10^(-7))) 
  tab[ii, "FDR"]<-sum(res[, "adj.P.Val"]<=0.05)
  tab[ii, "Sites"]<-nrow(res) 
  coef<-get(paste(studies[ii], ".coef", sep=""))
  tab[ii, "Min N"]<-range(coef[, "N.subjects"])[1] 
  tab[ii, "Max N"]<-range(coef[, "N.subjects"])[2] 
  p<-res[, "P.Value"]
  chisq<-qchisq(1-p,1)
  tab[ii, "Lambda"]<-round(median(chisq)/qchisq(0.5,1),4)
  rm(p, chisq, coef, res)
}

write.csv(tab, "PGC_EWAS_NRG1_PTSD_mod1.csv")

rm(list=ls())

########################################################################################
# Step 3: Prep Data
########################################################################################

load("PGC_EWAS_NRG1_PTSD_mod1.Rdata")

# The minimum adjusted p-values are:
min(DNHS.results[,"adj.P.Val"]) #  0.1603903
min(GTP.results[,"adj.P.Val"]) # 0.01499607

# Convert data.frames to matrices. This speeds up the later loops
str(DNHS.results) 
DNHS.results.m<-as.matrix(DNHS.results)
all(DNHS.results==DNHS.results.m)
DNHS.results<-DNHS.results.m
rm(DNHS.results.m)

str(GTP.results)
GTP.results.m<-as.matrix(GTP.results)
all(GTP.results==GTP.results.m)
GTP.results<-GTP.results.m
rm(GTP.results.m)

# Check that all the rownames 
sum(is.na(match(rownames(DNHS.coef), rownames(DNHS.results))))
sum(is.na(match(rownames(DNHS.coef), rownames(DNHS.ebayes))))
head(DNHS.results)
DNHS.results<-DNHS.results[rownames(DNHS.ebayes),]
head(DNHS.results)
head(DNHS.coef)
DNHS.coef<-DNHS.coef[rownames(DNHS.ebayes),]
head(DNHS.coef)
all(rownames(DNHS.results)==rownames(DNHS.coef))
all(rownames(DNHS.results)==rownames(DNHS.ebayes))

sum(is.na(match(rownames(GTP.coef), rownames(GTP.results))))
sum(is.na(match(rownames(GTP.coef), rownames(GTP.ebayes))))
head(GTP.results)
GTP.results<-GTP.results[rownames(GTP.ebayes),]
head(GTP.results)
head(GTP.coef)
GTP.coef<-GTP.coef[rownames(GTP.ebayes),]
head(GTP.coef)
all(rownames(GTP.results)==rownames(GTP.coef))
all(rownames(GTP.results)==rownames(GTP.ebayes))

# Determine the number of probes available in all studies
dnhs.sites<-rownames(DNHS.coef)
gtp.sites<-rownames(GTP.coef)
sites<-append(dnhs.sites, gtp.sites)
all<-intersect(dnhs.sites, gtp.sites)

sum(is.na(match(all, rownames(DNHS.coef)))) # these should all be 0 
sum(is.na(match(all, rownames(GTP.coef))))
length(all) 

rm(dnhs.sites, gtp.sites)

# Calculate one-sided p-values for each CpG site's t-statistic
DNHS.oneSided<-pt(DNHS.ebayes$t, df = (DNHS.ebayes$df.prior+DNHS.ebayes$df.residual))
all(rownames(DNHS.oneSided)==rownames(DNHS.results))
all(rownames(DNHS.oneSided)==rownames(DNHS.coef))
all(rownames(DNHS.oneSided)==rownames(DNHS.ebayes))

GTP.oneSided<-pt(GTP.ebayes$t, df = (GTP.ebayes$df.prior+GTP.ebayes$df.residual))
all(rownames(GTP.oneSided)==rownames(GTP.results))
all(rownames(GTP.oneSided)==rownames(GTP.coef))
all(rownames(GTP.oneSided)==rownames(GTP.ebayes))

# Subset intersecting sites

studies<-c("DNHS", "GTP")

for(ii in 1:length(studies)){
  coef<-get(paste(studies[ii], ".coef", sep=""))
  coef<-coef[all, ]
  assign(paste(studies[ii], ".coef", sep=""), coef)
  rm(coef)
  results<-get(paste(studies[ii], ".results", sep=""))
  results<-results[all, ]
  assign(paste(studies[ii], ".results", sep=""), results)
  rm(results)
  ebayes<-get(paste(studies[ii], ".ebayes", sep=""))
  ebayes<-ebayes[all, ]
  assign(paste(studies[ii], ".ebayes", sep=""), ebayes)
  rm(ebayes)
  oneSided<-get(paste(studies[ii], ".oneSided", sep=""))
  oneSided<-oneSided[all, ]
  assign(paste(studies[ii], ".oneSided", sep=""), oneSided)
  rm(oneSided)
}

save.image("PGC_EWAS_DataPrep_intersectSites_NRG1_PTSD_mod1.Rdata")

rm(list=ls())

########################################################################################
# Step 4: Meta-Analysis of Sites in All Studies
########################################################################################

load("PGC_EWAS_DataPrep_intersectSites_NRG1_PTSD_mod1.Rdata")

# Check that all rownames line up
all(rownames(DNHS.coef)==rownames(DNHS.results))
all(rownames(DNHS.coef)==rownames(DNHS.ebayes))
all(rownames(DNHS.coef)==names(DNHS.oneSided))

all(rownames(DNHS.coef)==rownames(GTP.coef))
all(rownames(DNHS.coef)==rownames(GTP.results))
all(rownames(DNHS.coef)==rownames(GTP.ebayes))
all(rownames(DNHS.coef)==names(GTP.oneSided))

pval.combined.marot<-NULL

for(ii in 1:nrow(DNHS.coef)){
  probe.subjs<-c(DNHS.coef[ii, "N.subjects"], GTP.coef[ii, "N.subjects"]) 
  weight<-sqrt(probe.subjs/sum(probe.subjs))
  DNHS.Zi<-qnorm(1-as.numeric(DNHS.oneSided[ii]));DNHS.Zi
  DNHS.Z<-DNHS.Zi*weight[1]; DNHS.Z # multiply by weight 1
  GTP.Zi<-qnorm(1-as.numeric(GTP.oneSided[ii]))
  GTP.Z<-GTP.Zi*weight[2] # multiply by weight 2
  Sg<-DNHS.Z+GTP.Z # add studies together
  p<-2*(1-pnorm(abs(Sg))) # calculated two-tailed p-value
  pval.combined.marot<-append(pval.combined.marot, p)
  if(ii%%10000==0){
    print(ii)
  }
}

pval.combined.marot<-cbind(rownames(DNHS.results), pval.combined.marot)

save(pval.combined.marot, file="PGC_EWAS_NRG1_PTSD_mod1_results.Rdata")

rm(list=ls())


```

\newpage

#### CA Main Effect Meta-Analysis

```{r, eval=F, echo=T}

rm(list=ls())

########################################################################################
# Step 1: Load Summary Statistics 
########################################################################################

# Step 1A: Load DNHS Results
load("DNHS_NRG1_PTSD_CM_mod1_10.21.16.Rdata")
results<-results.ca
fit2.ebayes<-fit2.ca.ebayes
rm(results.ca, fit2.ca, fit2.ca.ebayes)
DNHS.coef<-fit.coef
rm(fit.coef)
results<-results[order(results[, "P.Value"]), ]
rank<-c(1:nrow(results))
results<-cbind(results, rank)
DNHS.results<-results
rm(results, rank)
DNHS.ebayes<-fit2.ebayes
rm(fit2.ebayes)

# Step 1B: Load GTP Results
load("GTP_NRG1_PTSD_CM_mod1_03.08.17.Rdata")
results<-results.ca
fit2.ebayes<-fit2.ca.ebayes
rm(results.ca, fit2.ca, fit2.ca.ebayes)
GTP.coef<-fit.coef
rm(fit.coef)
results<-results[order(results[, "P.Value"]), ]
GTP.results<-results
rm(results)
rank<-c(1:nrow(GTP.results))
GTP.results<-cbind(GTP.results, rank)
rm(rank)
GTP.ebayes<-fit2.ebayes
rm(fit2.ebayes)

save.image("PGC_EWAS_NRG1_CA_mod1.Rdata")
rm(list=ls())

########################################################################################
# # Step 2: Summary of Results
########################################################################################

load("PGC_EWAS_NRG1_CA_mod1.Rdata")

studies<-c("DNHS", "GTP")
tab<-matrix(nrow=length(studies), ncol=9)
colnames(tab)<-c("Study", "Sites", "Min N", "Max N", "Lambda", "FDR", "p<5x10^-5",
                 "p<5x10^-6", "p<5x10^-7")

for(ii in 1:length(studies)){
  tab[ii, "Study"]<-studies[ii]
  res<-get(paste(studies[ii], ".results", sep=""))
  res<-res[!is.na(res[, "logFC"]),] 
  tab[ii, "p<5x10^-5"]<-sum(res[, "P.Value"]<(5*10^(-5))) 
  tab[ii, "p<5x10^-6"]<-sum(res[, "P.Value"]<(5*10^(-6))) 
  tab[ii, "p<5x10^-7"]<-sum(res[, "P.Value"]<(5*10^(-7))) 
  tab[ii, "FDR"]<-sum(res[, "adj.P.Val"]<=0.05)
  tab[ii, "Sites"]<-nrow(res) 
  coef<-get(paste(studies[ii], ".coef", sep=""))
  tab[ii, "Min N"]<-range(coef[, "N.subjects"])[1] 
  tab[ii, "Max N"]<-range(coef[, "N.subjects"])[2] 
  p<-res[, "P.Value"]
  chisq<-qchisq(1-p,1)
  tab[ii, "Lambda"]<-round(median(chisq)/qchisq(0.5,1),4)
  rm(p, chisq, coef, res)
}

write.csv(tab, "PGC_EWAS_NRG1_CA_mod1.csv")

rm(list=ls())

########################################################################################
# Step 3: Prep Data
########################################################################################

load("PGC_EWAS_NRG1_CA_mod1.Rdata")

# The minimum adjusted p-values are:
min(DNHS.results[,"adj.P.Val"]) #  0.1603903
min(GTP.results[,"adj.P.Val"]) # 0.01499607

# Convert data.frames to matrices. This speeds up the later loops
str(DNHS.results) 
DNHS.results.m<-as.matrix(DNHS.results)
all(DNHS.results==DNHS.results.m)
DNHS.results<-DNHS.results.m
rm(DNHS.results.m)

str(GTP.results)
GTP.results.m<-as.matrix(GTP.results)
all(GTP.results==GTP.results.m)
GTP.results<-GTP.results.m
rm(GTP.results.m)

# Check that all the rownames 
sum(is.na(match(rownames(DNHS.coef), rownames(DNHS.results))))
sum(is.na(match(rownames(DNHS.coef), rownames(DNHS.ebayes))))
head(DNHS.results)
DNHS.results<-DNHS.results[rownames(DNHS.ebayes),]
head(DNHS.results)
head(DNHS.coef)
DNHS.coef<-DNHS.coef[rownames(DNHS.ebayes),]
head(DNHS.coef)
all(rownames(DNHS.results)==rownames(DNHS.coef))
all(rownames(DNHS.results)==rownames(DNHS.ebayes))

sum(is.na(match(rownames(GTP.coef), rownames(GTP.results))))
sum(is.na(match(rownames(GTP.coef), rownames(GTP.ebayes))))
head(GTP.results)
GTP.results<-GTP.results[rownames(GTP.ebayes),]
head(GTP.results)
head(GTP.coef)
GTP.coef<-GTP.coef[rownames(GTP.ebayes),]
head(GTP.coef)
all(rownames(GTP.results)==rownames(GTP.coef))
all(rownames(GTP.results)==rownames(GTP.ebayes))

# Determine the number of probes available in all studies
dnhs.sites<-rownames(DNHS.coef)
gtp.sites<-rownames(GTP.coef)
sites<-append(dnhs.sites, gtp.sites)
all<-intersect(dnhs.sites, gtp.sites)

sum(is.na(match(all, rownames(DNHS.coef)))) # these should all be 0 
sum(is.na(match(all, rownames(GTP.coef))))
length(all) 

rm(dnhs.sites, gtp.sites)

# Calculate one-sided p-values for each CpG site's t-statistic
DNHS.oneSided<-pt(DNHS.ebayes$t, df = (DNHS.ebayes$df.prior+DNHS.ebayes$df.residual))
all(rownames(DNHS.oneSided)==rownames(DNHS.results))
all(rownames(DNHS.oneSided)==rownames(DNHS.coef))
all(rownames(DNHS.oneSided)==rownames(DNHS.ebayes))

GTP.oneSided<-pt(GTP.ebayes$t, df = (GTP.ebayes$df.prior+GTP.ebayes$df.residual))
all(rownames(GTP.oneSided)==rownames(GTP.results))
all(rownames(GTP.oneSided)==rownames(GTP.coef))
all(rownames(GTP.oneSided)==rownames(GTP.ebayes))

# Subset intersecting sites

studies<-c("DNHS", "GTP")

for(ii in 1:length(studies)){
  coef<-get(paste(studies[ii], ".coef", sep=""))
  coef<-coef[all, ]
  assign(paste(studies[ii], ".coef", sep=""), coef)
  rm(coef)
  results<-get(paste(studies[ii], ".results", sep=""))
  results<-results[all, ]
  assign(paste(studies[ii], ".results", sep=""), results)
  rm(results)
  ebayes<-get(paste(studies[ii], ".ebayes", sep=""))
  ebayes<-ebayes[all, ]
  assign(paste(studies[ii], ".ebayes", sep=""), ebayes)
  rm(ebayes)
  oneSided<-get(paste(studies[ii], ".oneSided", sep=""))
  oneSided<-oneSided[all, ]
  assign(paste(studies[ii], ".oneSided", sep=""), oneSided)
  rm(oneSided)
}

save.image("PGC_EWAS_DataPrep_intersectSites_NRG1_CA_mod1.Rdata")

rm(list=ls())

########################################################################################
# Step 4: Meta-Analysis of Sites in All Studies
########################################################################################

load("PGC_EWAS_DataPrep_intersectSites_NRG1_CA_mod1.Rdata")

# Check that all rownames line up
all(rownames(DNHS.coef)==rownames(DNHS.results))
all(rownames(DNHS.coef)==rownames(DNHS.ebayes))
all(rownames(DNHS.coef)==names(DNHS.oneSided))

all(rownames(DNHS.coef)==rownames(GTP.coef))
all(rownames(DNHS.coef)==rownames(GTP.results))
all(rownames(DNHS.coef)==rownames(GTP.ebayes))
all(rownames(DNHS.coef)==names(GTP.oneSided))

pval.combined.marot<-NULL

for(ii in 1:nrow(DNHS.coef)){
  probe.subjs<-c(DNHS.coef[ii, "N.subjects"], GTP.coef[ii, "N.subjects"]) 
  weight<-sqrt(probe.subjs/sum(probe.subjs))
  DNHS.Zi<-qnorm(1-as.numeric(DNHS.oneSided[ii]));DNHS.Zi
  DNHS.Z<-DNHS.Zi*weight[1]; DNHS.Z # multiply by weight 1
  GTP.Zi<-qnorm(1-as.numeric(GTP.oneSided[ii]))
  GTP.Z<-GTP.Zi*weight[2] # multiply by weight 2
  Sg<-DNHS.Z+GTP.Z # add studies together
  p<-2*(1-pnorm(abs(Sg))) # calculated two-tailed p-value
  pval.combined.marot<-append(pval.combined.marot, p)
  if(ii%%10000==0){
    print(ii)
  }
}

pval.combined.marot<-cbind(rownames(DNHS.results), pval.combined.marot)

save(pval.combined.marot, file="PGC_EWAS_NRG1_CA_mod1_results.Rdata")

rm(list=ls())
```

\newpage

### Model 2: 

#### PTSD Main Effect Meta-Analysis

```{r, eval=F, echo=T}

rm(list=ls())

########################################################################################
# Step 1: Load Summary Statistics 
########################################################################################

# Step 1A: Load DNHS Results
load("DNHS_NRG1_PTSD_CM_mod2_10.21.16.Rdata")
rm(results.ca, fit2.ca, fit2.ca.ebayes)
DNHS.coef<-fit.coef
rm(fit.coef)
results<-results[order(results[, "P.Value"]), ]
rank<-c(1:nrow(results))
results<-cbind(results, rank)
DNHS.results<-results
rm(results, rank)
DNHS.ebayes<-fit2.ebayes
rm(fit2.ebayes)

# Step 1B: Load GTP Results
load("GTP_NRG1_PTSD_CM_mod2_03.08.17.Rdata")
rm(results.ca, fit2.ca, fit2.ca.ebayes)
GTP.coef<-fit.coef
rm(fit.coef)
results<-results[order(results[, "P.Value"]), ]
GTP.results<-results
rm(results)
rank<-c(1:nrow(GTP.results))
GTP.results<-cbind(GTP.results, rank)
rm(rank)
GTP.ebayes<-fit2.ebayes
rm(fit2.ebayes)

save.image("PGC_EWAS_NRG1_PTSD_mod2.Rdata")
rm(list=ls())

########################################################################################
# # Step 2: Summary of Results
########################################################################################

load("PGC_EWAS_NRG1_PTSD_mod2.Rdata")

studies<-c("DNHS", "GTP")
tab<-matrix(nrow=length(studies), ncol=9)
colnames(tab)<-c("Study", "Sites", "Min N", "Max N", "Lambda", "FDR", "p<5x10^-5",
                 "p<5x10^-6", "p<5x10^-7")

for(ii in 1:length(studies)){
  tab[ii, "Study"]<-studies[ii]
  res<-get(paste(studies[ii], ".results", sep=""))
  res<-res[!is.na(res[, "logFC"]),] 
  tab[ii, "p<5x10^-5"]<-sum(res[, "P.Value"]<(5*10^(-5))) 
  tab[ii, "p<5x10^-6"]<-sum(res[, "P.Value"]<(5*10^(-6))) 
  tab[ii, "p<5x10^-7"]<-sum(res[, "P.Value"]<(5*10^(-7))) 
  tab[ii, "FDR"]<-sum(res[, "adj.P.Val"]<=0.05)
  tab[ii, "Sites"]<-nrow(res) 
  coef<-get(paste(studies[ii], ".coef", sep=""))
  tab[ii, "Min N"]<-range(coef[, "N.subjects"])[1] 
  tab[ii, "Max N"]<-range(coef[, "N.subjects"])[2] 
  p<-res[, "P.Value"]
  chisq<-qchisq(1-p,1)
  tab[ii, "Lambda"]<-round(median(chisq)/qchisq(0.5,1),4)
  rm(p, chisq, coef, res)
}

write.csv(tab, "PGC_EWAS_NRG1_PTSD_mod2.csv")

rm(list=ls())

########################################################################################
# Step 3: Prep Data
########################################################################################

load("PGC_EWAS_NRG1_PTSD_mod2.Rdata")

# The minimum adjusted p-values are:
min(DNHS.results[,"adj.P.Val"]) #  0.1603903
min(GTP.results[,"adj.P.Val"]) # 0.01499607

# Convert data.frames to matrices. This speeds up the later loops
str(DNHS.results) 
DNHS.results.m<-as.matrix(DNHS.results)
all(DNHS.results==DNHS.results.m)
DNHS.results<-DNHS.results.m
rm(DNHS.results.m)

str(GTP.results)
GTP.results.m<-as.matrix(GTP.results)
all(GTP.results==GTP.results.m)
GTP.results<-GTP.results.m
rm(GTP.results.m)

# Check that all the rownames 
sum(is.na(match(rownames(DNHS.coef), rownames(DNHS.results))))
sum(is.na(match(rownames(DNHS.coef), rownames(DNHS.ebayes))))
head(DNHS.results)
DNHS.results<-DNHS.results[rownames(DNHS.ebayes),]
head(DNHS.results)
head(DNHS.coef)
DNHS.coef<-DNHS.coef[rownames(DNHS.ebayes),]
head(DNHS.coef)
all(rownames(DNHS.results)==rownames(DNHS.coef))
all(rownames(DNHS.results)==rownames(DNHS.ebayes))

sum(is.na(match(rownames(GTP.coef), rownames(GTP.results))))
sum(is.na(match(rownames(GTP.coef), rownames(GTP.ebayes))))
head(GTP.results)
GTP.results<-GTP.results[rownames(GTP.ebayes),]
head(GTP.results)
head(GTP.coef)
GTP.coef<-GTP.coef[rownames(GTP.ebayes),]
head(GTP.coef)
all(rownames(GTP.results)==rownames(GTP.coef))
all(rownames(GTP.results)==rownames(GTP.ebayes))

# Determine the number of probes available in all studies
dnhs.sites<-rownames(DNHS.coef)
gtp.sites<-rownames(GTP.coef)
sites<-append(dnhs.sites, gtp.sites)
all<-intersect(dnhs.sites, gtp.sites)

sum(is.na(match(all, rownames(DNHS.coef)))) # these should all be 0 
sum(is.na(match(all, rownames(GTP.coef))))
length(all) 

rm(dnhs.sites, gtp.sites)

# Calculate one-sided p-values for each CpG site's t-statistic
DNHS.oneSided<-pt(DNHS.ebayes$t, df = (DNHS.ebayes$df.prior+DNHS.ebayes$df.residual))
all(rownames(DNHS.oneSided)==rownames(DNHS.results))
all(rownames(DNHS.oneSided)==rownames(DNHS.coef))
all(rownames(DNHS.oneSided)==rownames(DNHS.ebayes))

GTP.oneSided<-pt(GTP.ebayes$t, df = (GTP.ebayes$df.prior+GTP.ebayes$df.residual))
all(rownames(GTP.oneSided)==rownames(GTP.results))
all(rownames(GTP.oneSided)==rownames(GTP.coef))
all(rownames(GTP.oneSided)==rownames(GTP.ebayes))

# Subset intersecting sites

studies<-c("DNHS", "GTP")

for(ii in 1:length(studies)){
  coef<-get(paste(studies[ii], ".coef", sep=""))
  coef<-coef[all, ]
  assign(paste(studies[ii], ".coef", sep=""), coef)
  rm(coef)
  results<-get(paste(studies[ii], ".results", sep=""))
  results<-results[all, ]
  assign(paste(studies[ii], ".results", sep=""), results)
  rm(results)
  ebayes<-get(paste(studies[ii], ".ebayes", sep=""))
  ebayes<-ebayes[all, ]
  assign(paste(studies[ii], ".ebayes", sep=""), ebayes)
  rm(ebayes)
  oneSided<-get(paste(studies[ii], ".oneSided", sep=""))
  oneSided<-oneSided[all, ]
  assign(paste(studies[ii], ".oneSided", sep=""), oneSided)
  rm(oneSided)
}

save.image("PGC_EWAS_DataPrep_intersectSites_NRG1_PTSD_mod2.Rdata")

rm(list=ls())

########################################################################################
# Step 4: Meta-Analysis of Sites in All Studies
########################################################################################

load("PGC_EWAS_DataPrep_intersectSites_NRG1_PTSD_mod2.Rdata")

# Check that all rownames line up
all(rownames(DNHS.coef)==rownames(DNHS.results))
all(rownames(DNHS.coef)==rownames(DNHS.ebayes))
all(rownames(DNHS.coef)==names(DNHS.oneSided))

all(rownames(DNHS.coef)==rownames(GTP.coef))
all(rownames(DNHS.coef)==rownames(GTP.results))
all(rownames(DNHS.coef)==rownames(GTP.ebayes))
all(rownames(DNHS.coef)==names(GTP.oneSided))

pval.combined.marot<-NULL

for(ii in 1:nrow(DNHS.coef)){
  probe.subjs<-c(DNHS.coef[ii, "N.subjects"], GTP.coef[ii, "N.subjects"]) 
  weight<-sqrt(probe.subjs/sum(probe.subjs))
  DNHS.Zi<-qnorm(1-as.numeric(DNHS.oneSided[ii]));DNHS.Zi
  DNHS.Z<-DNHS.Zi*weight[1]; DNHS.Z # multiply by weight 1
  GTP.Zi<-qnorm(1-as.numeric(GTP.oneSided[ii]))
  GTP.Z<-GTP.Zi*weight[2] # multiply by weight 2
  Sg<-DNHS.Z+GTP.Z # add studies together
  p<-2*(1-pnorm(abs(Sg))) # calculated two-tailed p-value
  pval.combined.marot<-append(pval.combined.marot, p)
  if(ii%%10000==0){
    print(ii)
  }
}

pval.combined.marot<-cbind(rownames(DNHS.results), pval.combined.marot)

save(pval.combined.marot, file="PGC_EWAS_NRG1_PTSD_mod2_results.Rdata")

rm(list=ls())
```

\newpage

#### CA Main Effect Meta-Analysis

```{r, eval=F, echo=T}

rm(list=ls())

########################################################################################
# Step 1: Load Summary Statistics 
########################################################################################

# Step 1A: Load DNHS Results
load("DNHS_NRG1_PTSD_CM_mod2_10.21.16.Rdata")
results<-results.ca
fit2.ebayes<-fit2.ca.ebayes
rm(results.ca, fit2.ca, fit2.ca.ebayes)
DNHS.coef<-fit.coef
rm(fit.coef)
results<-results[order(results[, "P.Value"]), ]
rank<-c(1:nrow(results))
results<-cbind(results, rank)
DNHS.results<-results
rm(results, rank)
DNHS.ebayes<-fit2.ebayes
rm(fit2.ebayes)

# Step 1B: Load GTP Results
load("GTP_NRG1_PTSD_CM_mod2_03.08.17.Rdata")
results<-results.ca
fit2.ebayes<-fit2.ca.ebayes
rm(results.ca, fit2.ca, fit2.ca.ebayes)
GTP.coef<-fit.coef
rm(fit.coef)
results<-results[order(results[, "P.Value"]), ]
GTP.results<-results
rm(results)
rank<-c(1:nrow(GTP.results))
GTP.results<-cbind(GTP.results, rank)
rm(rank)
GTP.ebayes<-fit2.ebayes
rm(fit2.ebayes)

save.image("PGC_EWAS_NRG1_CA_mod2.Rdata")
rm(list=ls())

########################################################################################
# # Step 2: Summary of Results
########################################################################################

load("PGC_EWAS_NRG1_CA_mod2.Rdata")

studies<-c("DNHS", "GTP")
tab<-matrix(nrow=length(studies), ncol=9)
colnames(tab)<-c("Study", "Sites", "Min N", "Max N", "Lambda", "FDR", "p<5x10^-5",
                 "p<5x10^-6", "p<5x10^-7")

for(ii in 1:length(studies)){
  tab[ii, "Study"]<-studies[ii]
  res<-get(paste(studies[ii], ".results", sep=""))
  res<-res[!is.na(res[, "logFC"]),] 
  tab[ii, "p<5x10^-5"]<-sum(res[, "P.Value"]<(5*10^(-5))) 
  tab[ii, "p<5x10^-6"]<-sum(res[, "P.Value"]<(5*10^(-6))) 
  tab[ii, "p<5x10^-7"]<-sum(res[, "P.Value"]<(5*10^(-7))) 
  tab[ii, "FDR"]<-sum(res[, "adj.P.Val"]<=0.05)
  tab[ii, "Sites"]<-nrow(res) 
  coef<-get(paste(studies[ii], ".coef", sep=""))
  tab[ii, "Min N"]<-range(coef[, "N.subjects"])[1] 
  tab[ii, "Max N"]<-range(coef[, "N.subjects"])[2] 
  p<-res[, "P.Value"]
  chisq<-qchisq(1-p,1)
  tab[ii, "Lambda"]<-round(median(chisq)/qchisq(0.5,1),4)
  rm(p, chisq, coef, res)
}

write.csv(tab, "PGC_EWAS_NRG1_CA_mod2.csv")

rm(list=ls())

########################################################################################
# Step 3: Prep Data
########################################################################################

load("PGC_EWAS_NRG1_CA_mod2.Rdata")

# The minimum adjusted p-values are:
min(DNHS.results[,"adj.P.Val"]) #  0.1603903
min(GTP.results[,"adj.P.Val"]) # 0.01499607

# Convert data.frames to matrices. This speeds up the later loops
str(DNHS.results) 
DNHS.results.m<-as.matrix(DNHS.results)
all(DNHS.results==DNHS.results.m)
DNHS.results<-DNHS.results.m
rm(DNHS.results.m)

str(GTP.results)
GTP.results.m<-as.matrix(GTP.results)
all(GTP.results==GTP.results.m)
GTP.results<-GTP.results.m
rm(GTP.results.m)

# Check that all the rownames 
sum(is.na(match(rownames(DNHS.coef), rownames(DNHS.results))))
sum(is.na(match(rownames(DNHS.coef), rownames(DNHS.ebayes))))
head(DNHS.results)
DNHS.results<-DNHS.results[rownames(DNHS.ebayes),]
head(DNHS.results)
head(DNHS.coef)
DNHS.coef<-DNHS.coef[rownames(DNHS.ebayes),]
head(DNHS.coef)
all(rownames(DNHS.results)==rownames(DNHS.coef))
all(rownames(DNHS.results)==rownames(DNHS.ebayes))

sum(is.na(match(rownames(GTP.coef), rownames(GTP.results))))
sum(is.na(match(rownames(GTP.coef), rownames(GTP.ebayes))))
head(GTP.results)
GTP.results<-GTP.results[rownames(GTP.ebayes),]
head(GTP.results)
head(GTP.coef)
GTP.coef<-GTP.coef[rownames(GTP.ebayes),]
head(GTP.coef)
all(rownames(GTP.results)==rownames(GTP.coef))
all(rownames(GTP.results)==rownames(GTP.ebayes))

# Determine the number of probes available in all studies
dnhs.sites<-rownames(DNHS.coef)
gtp.sites<-rownames(GTP.coef)
sites<-append(dnhs.sites, gtp.sites)
all<-intersect(dnhs.sites, gtp.sites)

sum(is.na(match(all, rownames(DNHS.coef)))) # these should all be 0 
sum(is.na(match(all, rownames(GTP.coef))))
length(all) 

rm(dnhs.sites, gtp.sites)

# Calculate one-sided p-values for each CpG site's t-statistic
DNHS.oneSided<-pt(DNHS.ebayes$t, df = (DNHS.ebayes$df.prior+DNHS.ebayes$df.residual))
all(rownames(DNHS.oneSided)==rownames(DNHS.results))
all(rownames(DNHS.oneSided)==rownames(DNHS.coef))
all(rownames(DNHS.oneSided)==rownames(DNHS.ebayes))

GTP.oneSided<-pt(GTP.ebayes$t, df = (GTP.ebayes$df.prior+GTP.ebayes$df.residual))
all(rownames(GTP.oneSided)==rownames(GTP.results))
all(rownames(GTP.oneSided)==rownames(GTP.coef))
all(rownames(GTP.oneSided)==rownames(GTP.ebayes))

# Subset intersecting sites

studies<-c("DNHS", "GTP")

for(ii in 1:length(studies)){
  coef<-get(paste(studies[ii], ".coef", sep=""))
  coef<-coef[all, ]
  assign(paste(studies[ii], ".coef", sep=""), coef)
  rm(coef)
  results<-get(paste(studies[ii], ".results", sep=""))
  results<-results[all, ]
  assign(paste(studies[ii], ".results", sep=""), results)
  rm(results)
  ebayes<-get(paste(studies[ii], ".ebayes", sep=""))
  ebayes<-ebayes[all, ]
  assign(paste(studies[ii], ".ebayes", sep=""), ebayes)
  rm(ebayes)
  oneSided<-get(paste(studies[ii], ".oneSided", sep=""))
  oneSided<-oneSided[all, ]
  assign(paste(studies[ii], ".oneSided", sep=""), oneSided)
  rm(oneSided)
}

save.image("PGC_EWAS_DataPrep_intersectSites_NRG1_CA_mod2.Rdata")

rm(list=ls())

########################################################################################
# Step 4: Meta-Analysis of Sites in All Studies
########################################################################################

load("PGC_EWAS_DataPrep_intersectSites_NRG1_CA_mod2.Rdata")

# Check that all rownames line up
all(rownames(DNHS.coef)==rownames(DNHS.results))
all(rownames(DNHS.coef)==rownames(DNHS.ebayes))
all(rownames(DNHS.coef)==names(DNHS.oneSided))

all(rownames(DNHS.coef)==rownames(GTP.coef))
all(rownames(DNHS.coef)==rownames(GTP.results))
all(rownames(DNHS.coef)==rownames(GTP.ebayes))
all(rownames(DNHS.coef)==names(GTP.oneSided))

pval.combined.marot<-NULL

for(ii in 1:nrow(DNHS.coef)){
  probe.subjs<-c(DNHS.coef[ii, "N.subjects"], GTP.coef[ii, "N.subjects"]) 
  weight<-sqrt(probe.subjs/sum(probe.subjs))
  DNHS.Zi<-qnorm(1-as.numeric(DNHS.oneSided[ii]));DNHS.Zi
  DNHS.Z<-DNHS.Zi*weight[1]; DNHS.Z # multiply by weight 1
  GTP.Zi<-qnorm(1-as.numeric(GTP.oneSided[ii]))
  GTP.Z<-GTP.Zi*weight[2] # multiply by weight 2
  Sg<-DNHS.Z+GTP.Z # add studies together
  p<-2*(1-pnorm(abs(Sg))) # calculated two-tailed p-value
  pval.combined.marot<-append(pval.combined.marot, p)
  if(ii%%10000==0){
    print(ii)
  }
}

pval.combined.marot<-cbind(rownames(DNHS.results), pval.combined.marot)

save(pval.combined.marot, file="PGC_EWAS_NRG1_CA_mod2_results.Rdata")

rm(list=ls())
```

\newpage

#### PTSDxCA Interaction Effect Meta-Analysis

```{r, eval=F, echo=T}

rm(list=ls())

########################################################################################
# Step 1: Load Summary Statistics 
########################################################################################

# Step 1A: Load DNHS Results
load("DNHS_NRG1_PTSD_CM_mod2_10.21.16.Rdata")
results<-results.caPTSD
fit2.ebayes<-fit2.caPTSD.ebayes
rm(results.ca, fit2.ca, fit2.ca.ebayes, fit2.caPTSD, fit2.caPTSD.ebayes, results.caPTSD)
DNHS.coef<-fit.coef
rm(fit.coef)
results<-results[order(results[, "P.Value"]), ]
rank<-c(1:nrow(results))
results<-cbind(results, rank)
DNHS.results<-results
rm(results, rank)
DNHS.ebayes<-fit2.ebayes
rm(fit2.ebayes)

# Step 1B: Load GTP Results
load("GTP_NRG1_PTSD_CM_mod2_03.08.17.Rdata")
results<-results.caPTSD
fit2.ebayes<-fit2.caPTSD.ebayes
rm(results.ca, fit2.ca, fit2.ca.ebayes, fit2.caPTSD, fit2.caPTSD.ebayes, results.caPTSD)
GTP.coef<-fit.coef
rm(fit.coef)
results<-results[order(results[, "P.Value"]), ]
GTP.results<-results
rm(results)
rank<-c(1:nrow(GTP.results))
GTP.results<-cbind(GTP.results, rank)
rm(rank)
GTP.ebayes<-fit2.ebayes
rm(fit2.ebayes)

save.image("PGC_EWAS_NRG1_PTSDxCA_mod2.Rdata")
rm(list=ls())

########################################################################################
# # Step 2: Summary of Results
########################################################################################

load("PGC_EWAS_NRG1_PTSDxCA_mod2.Rdata")

studies<-c("DNHS", "GTP")
tab<-matrix(nrow=length(studies), ncol=9)
colnames(tab)<-c("Study", "Sites", "Min N", "Max N", "Lambda", "FDR", "p<5x10^-5",
                 "p<5x10^-6", "p<5x10^-7")

for(ii in 1:length(studies)){
  tab[ii, "Study"]<-studies[ii]
  res<-get(paste(studies[ii], ".results", sep=""))
  res<-res[!is.na(res[, "logFC"]),] 
  tab[ii, "p<5x10^-5"]<-sum(res[, "P.Value"]<(5*10^(-5))) 
  tab[ii, "p<5x10^-6"]<-sum(res[, "P.Value"]<(5*10^(-6))) 
  tab[ii, "p<5x10^-7"]<-sum(res[, "P.Value"]<(5*10^(-7))) 
  tab[ii, "FDR"]<-sum(res[, "adj.P.Val"]<=0.05)
  tab[ii, "Sites"]<-nrow(res) 
  coef<-get(paste(studies[ii], ".coef", sep=""))
  tab[ii, "Min N"]<-range(coef[, "N.subjects"])[1] 
  tab[ii, "Max N"]<-range(coef[, "N.subjects"])[2] 
  p<-res[, "P.Value"]
  chisq<-qchisq(1-p,1)
  tab[ii, "Lambda"]<-round(median(chisq)/qchisq(0.5,1),4)
  rm(p, chisq, coef, res)
}

write.csv(tab, "PGC_EWAS_NRG1_PTSDxCA_mod2.csv")

rm(list=ls())

########################################################################################
# Step 3: Prep Data
########################################################################################

load("PGC_EWAS_NRG1_PTSDxCA_mod2.Rdata")

# The minimum adjusted p-values are:
min(DNHS.results[,"adj.P.Val"]) #  0.1603903
min(GTP.results[,"adj.P.Val"]) # 0.01499607

# Convert data.frames to matrices. This speeds up the later loops
str(DNHS.results) 
DNHS.results.m<-as.matrix(DNHS.results)
all(DNHS.results==DNHS.results.m)
DNHS.results<-DNHS.results.m
rm(DNHS.results.m)

str(GTP.results)
GTP.results.m<-as.matrix(GTP.results)
all(GTP.results==GTP.results.m)
GTP.results<-GTP.results.m
rm(GTP.results.m)

# Check that all the rownames 
sum(is.na(match(rownames(DNHS.coef), rownames(DNHS.results))))
sum(is.na(match(rownames(DNHS.coef), rownames(DNHS.ebayes))))
head(DNHS.results)
DNHS.results<-DNHS.results[rownames(DNHS.ebayes),]
head(DNHS.results)
head(DNHS.coef)
DNHS.coef<-DNHS.coef[rownames(DNHS.ebayes),]
head(DNHS.coef)
all(rownames(DNHS.results)==rownames(DNHS.coef))
all(rownames(DNHS.results)==rownames(DNHS.ebayes))

sum(is.na(match(rownames(GTP.coef), rownames(GTP.results))))
sum(is.na(match(rownames(GTP.coef), rownames(GTP.ebayes))))
head(GTP.results)
GTP.results<-GTP.results[rownames(GTP.ebayes),]
head(GTP.results)
head(GTP.coef)
GTP.coef<-GTP.coef[rownames(GTP.ebayes),]
head(GTP.coef)
all(rownames(GTP.results)==rownames(GTP.coef))
all(rownames(GTP.results)==rownames(GTP.ebayes))

# Determine the number of probes available in all studies
dnhs.sites<-rownames(DNHS.coef)
gtp.sites<-rownames(GTP.coef)
sites<-append(dnhs.sites, gtp.sites)
all<-intersect(dnhs.sites, gtp.sites)

sum(is.na(match(all, rownames(DNHS.coef)))) # these should all be 0 
sum(is.na(match(all, rownames(GTP.coef))))
length(all) 

rm(dnhs.sites, gtp.sites)

# Calculate one-sided p-values for each CpG site's t-statistic
DNHS.oneSided<-pt(DNHS.ebayes$t, df = (DNHS.ebayes$df.prior+DNHS.ebayes$df.residual))
all(rownames(DNHS.oneSided)==rownames(DNHS.results))
all(rownames(DNHS.oneSided)==rownames(DNHS.coef))
all(rownames(DNHS.oneSided)==rownames(DNHS.ebayes))

GTP.oneSided<-pt(GTP.ebayes$t, df = (GTP.ebayes$df.prior+GTP.ebayes$df.residual))
all(rownames(GTP.oneSided)==rownames(GTP.results))
all(rownames(GTP.oneSided)==rownames(GTP.coef))
all(rownames(GTP.oneSided)==rownames(GTP.ebayes))

# Subset intersecting sites

studies<-c("DNHS", "GTP")

for(ii in 1:length(studies)){
  coef<-get(paste(studies[ii], ".coef", sep=""))
  coef<-coef[all, ]
  assign(paste(studies[ii], ".coef", sep=""), coef)
  rm(coef)
  results<-get(paste(studies[ii], ".results", sep=""))
  results<-results[all, ]
  assign(paste(studies[ii], ".results", sep=""), results)
  rm(results)
  ebayes<-get(paste(studies[ii], ".ebayes", sep=""))
  ebayes<-ebayes[all, ]
  assign(paste(studies[ii], ".ebayes", sep=""), ebayes)
  rm(ebayes)
  oneSided<-get(paste(studies[ii], ".oneSided", sep=""))
  oneSided<-oneSided[all, ]
  assign(paste(studies[ii], ".oneSided", sep=""), oneSided)
  rm(oneSided)
}

save.image("PGC_EWAS_DataPrep_intersectSites_NRG1_PTSDxCA_mod2.Rdata")

rm(list=ls())

########################################################################################
# Step 4: Meta-Analysis of Sites in All Studies
########################################################################################

load("PGC_EWAS_DataPrep_intersectSites_NRG1_PTSDxCA_mod2.Rdata")

# Check that all rownames line up
all(rownames(DNHS.coef)==rownames(DNHS.results))
all(rownames(DNHS.coef)==rownames(DNHS.ebayes))
all(rownames(DNHS.coef)==names(DNHS.oneSided))

all(rownames(DNHS.coef)==rownames(GTP.coef))
all(rownames(DNHS.coef)==rownames(GTP.results))
all(rownames(DNHS.coef)==rownames(GTP.ebayes))
all(rownames(DNHS.coef)==names(GTP.oneSided))

pval.combined.marot<-NULL

for(ii in 1:nrow(DNHS.coef)){
  probe.subjs<-c(DNHS.coef[ii, "N.subjects"], GTP.coef[ii, "N.subjects"]) 
  weight<-sqrt(probe.subjs/sum(probe.subjs))
  DNHS.Zi<-qnorm(1-as.numeric(DNHS.oneSided[ii]));DNHS.Zi
  DNHS.Z<-DNHS.Zi*weight[1]; DNHS.Z # multiply by weight 1
  GTP.Zi<-qnorm(1-as.numeric(GTP.oneSided[ii]))
  GTP.Z<-GTP.Zi*weight[2] # multiply by weight 2
  Sg<-DNHS.Z+GTP.Z # add studies together
  p<-2*(1-pnorm(abs(Sg))) # calculated two-tailed p-value
  pval.combined.marot<-append(pval.combined.marot, p)
  if(ii%%10000==0){
    print(ii)
  }
}

pval.combined.marot<-cbind(rownames(DNHS.results), pval.combined.marot)

save(pval.combined.marot, file="PGC_EWAS_NRG1_PTSDxCA_mod2_results.Rdata")

rm(list=ls())
```


## Summary Scripts:

### Model 1: Summary

```{r, eval=F, echo=T}
rm(list=ls())

########################################################################################
# Summarize All Results
########################################################################################

cpg<-"cg23637605"
tab<-matrix(nrow=13, ncol=3)
colnames(tab)<-c("DNHS", "GTP", "Meta")
rownames(tab)<-c("bPTSD", "sePTSD", "pPTSD", "FDR_PTSD", "p_1sPTSD", "lambdaPTSD",
                 "bCA", "seCA", "pCA", "FDR_CA", "p_1sCA", "lambdaCA", "N")
class(tab)<-"numeric"

# DNHS
load("DNHS_NRG1_PTSD_CM_mod1_10.21.16.Rdata")
ptsd.oneSided<-pt(fit2.ebayes$t, df = (fit2.ebayes$df.prior+fit2.ebayes$df.residual))
ca.oneSided<-pt(fit2.ca.ebayes$t, df = (fit2.ca.ebayes$df.prior+fit2.ca.ebayes$df.residual))

tab["bPTSD", "DNHS"]<-fit.coef[cpg, "PTSDpm"]
tab["sePTSD", "DNHS"]<-results[cpg, "logFC"]/results[cpg, "t"]
tab["pPTSD", "DNHS"]<-results[cpg, "P.Value"]
tab["FDR_PTSD", "DNHS"]<-results[cpg, "adj.P.Val"]
tab["p_1sPTSD", "DNHS"]<-ptsd.oneSided[cpg,]
pvalue <- as.numeric(results[, "P.Value"])
chisq <- qchisq(1-pvalue,1)
tab["lambdaPTSD", "DNHS"]<-median(chisq)/qchisq(0.5,1)
rm(pvalue, chisq)
tab["N", "DNHS"]<-fit.coef[cpg, "N.subjects"]

tab["bCA", "DNHS"]<-fit.coef[cpg, "CA_ordinary"]
tab["seCA", "DNHS"]<-results.ca[cpg, "logFC"]/results.ca[cpg, "t"]
tab["pCA", "DNHS"]<-results.ca[cpg, "P.Value"]
tab["FDR_CA", "DNHS"]<-results.ca[cpg, "adj.P.Val"]
tab["p_1sCA", "DNHS"]<-ca.oneSided[cpg,]
pvalue <- as.numeric(results.ca[, "P.Value"])
chisq <- qchisq(1-pvalue,1)
tab["lambdaCA", "DNHS"]<-median(chisq)/qchisq(0.5,1)
rm(pvalue, chisq)

rm(list=ls()[-match(c("tab", "cpg"), ls())])

# GTP
load("GTP_NRG1_PTSD_CM_mod1_03.08.17.Rdata")
ptsd.oneSided<-pt(fit2.ebayes$t, df = (fit2.ebayes$df.prior+fit2.ebayes$df.residual))
ca.oneSided<-pt(fit2.ca.ebayes$t, df = (fit2.ca.ebayes$df.prior+fit2.ca.ebayes$df.residual))

tab["bPTSD", "GTP"]<-fit.coef[cpg, "PTSDcurr"]
tab["sePTSD", "GTP"]<-results[cpg, "logFC"]/results[cpg, "t"]
tab["pPTSD", "GTP"]<-results[cpg, "P.Value"]
tab["FDR_PTSD", "GTP"]<-results[cpg, "adj.P.Val"]
tab["p_1sPTSD", "GTP"]<-ptsd.oneSided[cpg,]
pvalue <- as.numeric(results[, "P.Value"])
chisq <- qchisq(1-pvalue,1)
tab["lambdaPTSD", "GTP"]<-median(chisq)/qchisq(0.5,1)
rm(pvalue, chisq)
tab["N", "GTP"]<-fit.coef[cpg, "N.subjects"]

tab["bCA", "GTP"]<-fit.coef[cpg, "CA_ordinary"]
tab["seCA", "GTP"]<-results.ca[cpg, "logFC"]/results.ca[cpg, "t"]
tab["pCA", "GTP"]<-results.ca[cpg, "P.Value"]
tab["FDR_CA", "GTP"]<-results.ca[cpg, "adj.P.Val"]
tab["p_1sCA", "GTP"]<-ca.oneSided[cpg,]
pvalue <- as.numeric(results.ca[, "P.Value"])
chisq <- qchisq(1-pvalue,1)
tab["lambdaCA", "GTP"]<-median(chisq)/qchisq(0.5,1)
rm(pvalue, chisq)

rm(list=ls()[-match(c("tab", "cpg"), ls())])

# Meta-Analysis

ggd.qqplot = function(pvector, main=NULL, ...) {
  o = -log10(sort(pvector,decreasing=F))
  e = -log10( 1:length(o)/length(o) )
  plot(e,o,pch=19,cex=2, cex.lab=4.5, cex.axis=4, main=main, ...,
       xlab=expression(Expected~~-log[10](italic(p))),
       ylab=expression(Observed~~-log[10](italic(p))),
       xlim=c(0,max(e)), ylim=c(0,max(o)))
  lines(e,e,col="red")
}

load("PGC_EWAS_NRG1_PTSD_mod1_results.Rdata")
res<-pval.combined.marot
colnames(res)<-c("CpG", "p")
FDR<-p.adjust(as.numeric(res[, "p"]), method="fdr", n=nrow(res))
res<-cbind(res, FDR)
rownames(res)<-res[, "CpG"]
tab["pPTSD", "Meta"]<-res[cpg, "p"]
tab["FDR_PTSD", "Meta"]<-res[cpg, "FDR"]
pvalue <- as.numeric(res[, "p"])
chisq <- qchisq(1-pvalue,1)
tab["lambdaPTSD", "Meta"]<-median(chisq)/qchisq(0.5,1)

png("PGC_EWAS_NRG1_PTSD_mod1_QQplot.png", width=1280, height=960, units="px", bg="white")
par(mar=c(10,10,4,2), mgp=c(6,2,0))
ggd.qqplot(pvalue, main = paste("Model 1 PTSD QQ plot"))
dev.off()

rm(list=ls()[-match(c("tab", "cpg", "ggd.qqplot"), ls())])

load("PGC_EWAS_NRG1_CA_mod1_results.Rdata")
res<-pval.combined.marot
colnames(res)<-c("CpG", "p")
FDR<-p.adjust(as.numeric(res[, "p"]), method="fdr", n=nrow(res))
res<-cbind(res, FDR)
rownames(res)<-res[, "CpG"]
tab["pCA", "Meta"]<-res[cpg, "p"]
tab["FDR_CA", "Meta"]<-res[cpg, "FDR"]
pvalue <- as.numeric(res[, "p"])
chisq <- qchisq(1-pvalue,1)
tab["lambdaCA", "Meta"]<-median(chisq)/qchisq(0.5,1)

png("PGC_EWAS_NRG1_CA_mod1_QQplot.png", width=1280, height=960, units="px", bg="white")
par(mar=c(10,10,4,2), mgp=c(6,2,0))
ggd.qqplot(pvalue, main = paste("Model 1 CM QQ plot"))
dev.off()

rm(list=ls()[-match(c("tab", "cpg"), ls())])

# Coefficients
betas<-as.numeric(tab["bPTSD", ])
betas<-betas[!is.na(betas)]
stderr<-as.numeric(tab["sePTSD", ])
stderr<-stderr[!is.na(stderr)]
weights<-1/(stderr^2)
tab["bPTSD", "Meta"]<-sum(betas*weights)/sum(weights)
tab["sePTSD", "Meta"]<-1/sum(weights)
rm(betas, stderr, weights)

betas<-as.numeric(tab["bCA", ])
betas<-betas[!is.na(betas)]
stderr<-as.numeric(tab["seCA", ])
stderr<-stderr[!is.na(stderr)]
weights<-1/(stderr^2)
tab["bCA", "Meta"]<-sum(betas*weights)/sum(weights)
tab["seCA", "Meta"]<-1/sum(weights)
rm(betas, stderr, weights)
class(tab)<-"numeric"
tab["N", "Meta"]<-sum(tab["N", ], na.rm=T)

write.csv(tab, paste("PGC_EWAS_NRG1_PTSD_mod1_", cpg, ".csv", sep=""))

# PTSD Forest Plot
betas<-tab["bPTSD", c("DNHS", "GTP")]
stderr<-tab["sePTSD", c("DNHS", "GTP")]
lower<-betas-1.96*stderr
upper<-betas+1.96*stderr
betaC<-tab["bPTSD", "Meta"]
stderrC<-tab["sePTSD", "Meta"]
lowerC<-betaC-(1.96*stderrC)
upperC<-betaC+(1.96*stderrC)
beta.table<-c("", "Beta", round(betas,3), NA, round(betaC,3))
subjs<-tab["N", c("DNHS", "GTP")]
Ns<-as.character(unlist(c("", "N", subjs, NA, sum(subjs))))

studies<-c("DNHS", "GTP")
betas<-c(NA, NA, round(betas,3), NA, round(betaC,3))
lower<-c(NA, NA, lower, NA, lowerC)
upper<-c(NA, NA, upper, NA, upperC)
stud<-c(cpg, "Study", studies, NA, "Summary")
summary<-cbind(stud, beta.table, Ns)
summary[which(is.na(summary))]<-""
rownames(summary)<-c(1:nrow(summary))
cochrane<-data.frame(betas, lower, upper)

png(paste("PGC_EWAS_NRG1_PTSD_mod1_", cpg, ".png", sep=""), width=960, height=640,units="px")
forestplot(summary, cochrane,
           boxsize=c(1, 1, 0.2, 0.3), lwd.ci=1.5,
           new_page = TRUE,
           is.summary=c(TRUE,TRUE,rep(FALSE,2),TRUE),
           xlog=FALSE, #lineheight=unit(1,"cm"),
           col=fpColors(box="black",line="black", summary="royalblue"),
           xticks=(c(-0.4, -0.2, 0.1)),
           txt_gp = fpTxtGp( label = list(gpar(fontfamily="", cex=3)),
                             ticks = gpar(fontfamily = "", cex=2)))
dev.off()
rm(list=ls()[-match(c("tab", "cpg"), ls())])

# CA Forest Plot
betas<-tab["bCA", c("DNHS", "GTP")]
stderr<-tab["seCA", c("DNHS", "GTP")]
lower<-betas-1.96*stderr
upper<-betas+1.96*stderr
betaC<-tab["bCA", "Meta"]
stderrC<-tab["seCA", "Meta"]
lowerC<-betaC-(1.96*stderrC)
upperC<-betaC+(1.96*stderrC)
beta.table<-c("", "Beta", round(betas,3), NA, round(betaC,3))
subjs<-tab["N", c("DNHS", "GTP")]
Ns<-as.character(unlist(c("", "N", subjs, NA, sum(subjs))))

studies<-c("DNHS", "GTP")
betas<-c(NA, NA, round(betas,3), NA, round(betaC,3))
lower<-c(NA, NA, lower, NA, lowerC)
upper<-c(NA, NA, upper, NA, upperC)
stud<-c(cpg, "Study", studies, NA, "Summary")
summary<-cbind(stud, beta.table, Ns)
summary[which(is.na(summary))]<-""
rownames(summary)<-c(1:nrow(summary))

cochrane<-data.frame(betas, lower, upper)
png(paste("PGC_EWAS_NRG1_CA_mod1_", cpg, ".png", sep=""), width=960, height=640,units="px")
forestplot(summary, cochrane,
           boxsize=c(1, 1, 0.2, 0.3), lwd.ci=1.5,
           new_page = TRUE,
           is.summary=c(TRUE,TRUE,rep(FALSE,2),TRUE),
           xlog=FALSE, #lineheight=unit(1,"cm"),
           col=fpColors(box="black",line="black", summary="royalblue"),
           xticks=(c(-0.4, -0.2, 0.1)),
           txt_gp = fpTxtGp( label = list(gpar(fontfamily="", cex=3)),
                             ticks = gpar(fontfamily = "", cex=2)))
dev.off()
```

### Model 2: Results


```{r, eval=F, echo=T}
rm(list=ls())

########################################################################################
# Summarize All Results
########################################################################################

cpg<-"cg23637605"
tab<-matrix(nrow=19, ncol=3)
colnames(tab)<-c("DNHS", "GTP", "Meta")
rownames(tab)<-c("bPTSD", "sePTSD", "pPTSD", "FDR_PTSD", "p_1sPTSD", "lambdaPTSD",
                 "bCA", "seCA", "pCA", "FDR_CA", "p_1sCA", "lambdaCA",
                 "bPTSDxCA", "sePTSDxCA", "pPTSDxCA", "FDR_PTSDxCA", 
                 "p_1sPTSDxCA", "lambdaPTSDxCA", "N")

# DNHS
load("DNHS_NRG1_PTSD_CM_mod2_10.21.16.Rdata")
ptsd.oneSided<-pt(fit2.ebayes$t, df = (fit2.ebayes$df.prior+fit2.ebayes$df.residual))
ca.oneSided<-pt(fit2.ca.ebayes$t, df = (fit2.ca.ebayes$df.prior+fit2.ca.ebayes$df.residual))
ptsdxca.oneSided<-pt(fit2.caPTSD.ebayes$t, 
                     df = (fit2.caPTSD.ebayes$df.prior+fit2.caPTSD.ebayes$df.residual))

tab["bPTSD", "DNHS"]<-fit.coef[cpg, "PTSDpm"]
tab["sePTSD", "DNHS"]<-results[cpg, "logFC"]/results[cpg, "t"]
tab["pPTSD", "DNHS"]<-results[cpg, "P.Value"]
tab["FDR_PTSD", "DNHS"]<-results[cpg, "adj.P.Val"]
tab["p_1sPTSD", "DNHS"]<-ptsd.oneSided[cpg,]
pvalue <- as.numeric(results[, "P.Value"])
chisq <- qchisq(1-pvalue,1)
tab["lambdaPTSD", "DNHS"]<-median(chisq)/qchisq(0.5,1)
rm(pvalue, chisq)
tab["N", "DNHS"]<-fit.coef[cpg, "N.subjects"]

tab["bCA", "DNHS"]<-fit.coef[cpg, "CA_ordinary"]
tab["seCA", "DNHS"]<-results.ca[cpg, "logFC"]/results.ca[cpg, "t"]
tab["pCA", "DNHS"]<-results.ca[cpg, "P.Value"]
tab["FDR_CA", "DNHS"]<-results.ca[cpg, "adj.P.Val"]
tab["p_1sCA", "DNHS"]<-ca.oneSided[cpg,]
pvalue <- as.numeric(results.ca[, "P.Value"])
chisq <- qchisq(1-pvalue,1)
tab["lambdaCA", "DNHS"]<-median(chisq)/qchisq(0.5,1)
rm(pvalue, chisq)

tab["bPTSDxCA", "DNHS"]<-fit.coef[cpg, "PTSDpmXCA_ordinary"]
tab["sePTSDxCA", "DNHS"]<-results.caPTSD[cpg, "logFC"]/results.caPTSD[cpg, "t"]
tab["pPTSDxCA", "DNHS"]<-results.caPTSD[cpg, "P.Value"]
tab["FDR_PTSDxCA", "DNHS"]<-results.caPTSD[cpg, "adj.P.Val"]
tab["p_1sPTSDxCA", "DNHS"]<-ptsdxca.oneSided[cpg,]
pvalue <- as.numeric(results.caPTSD[, "P.Value"])
chisq <- qchisq(1-pvalue,1)
tab["lambdaPTSDxCA", "DNHS"]<-median(chisq)/qchisq(0.5,1)
rm(pvalue, chisq)

rm(list=ls()[-match(c("tab", "cpg"), ls())])

# GTP
load("GTP_NRG1_PTSD_CM_mod2_03.08.17.Rdata")
ptsd.oneSided<-pt(fit2.ebayes$t, df = (fit2.ebayes$df.prior+fit2.ebayes$df.residual))
ca.oneSided<-pt(fit2.ca.ebayes$t, df = (fit2.ca.ebayes$df.prior+fit2.ca.ebayes$df.residual))
ptsdxca.oneSided<-pt(fit2.caPTSD.ebayes$t, 
                     df = (fit2.caPTSD.ebayes$df.prior+fit2.caPTSD.ebayes$df.residual))

tab["bPTSD", "GTP"]<-fit.coef[cpg, "PTSDcurr"]
tab["sePTSD", "GTP"]<-results[cpg, "logFC"]/results[cpg, "t"]
tab["pPTSD", "GTP"]<-results[cpg, "P.Value"]
tab["FDR_PTSD", "GTP"]<-results[cpg, "adj.P.Val"]
tab["p_1sPTSD", "GTP"]<-ptsd.oneSided[cpg,]
pvalue <- as.numeric(results[, "P.Value"])
chisq <- qchisq(1-pvalue,1)
tab["lambdaPTSD", "GTP"]<-median(chisq)/qchisq(0.5,1)
rm(pvalue, chisq)
tab["N", "GTP"]<-fit.coef[cpg, "N.subjects"]

tab["bCA", "GTP"]<-fit.coef[cpg, "CA_ordinary"]
tab["seCA", "GTP"]<-results.ca[cpg, "logFC"]/results.ca[cpg, "t"]
tab["pCA", "GTP"]<-results.ca[cpg, "P.Value"]
tab["FDR_CA", "GTP"]<-results.ca[cpg, "adj.P.Val"]
tab["p_1sCA", "GTP"]<-ca.oneSided[cpg,]
pvalue <- as.numeric(results.ca[, "P.Value"])
chisq <- qchisq(1-pvalue,1)
tab["lambdaCA", "GTP"]<-median(chisq)/qchisq(0.5,1)
rm(pvalue, chisq)

tab["bPTSDxCA", "GTP"]<-fit.coef[cpg, "PTSDcurrXCA_ordinary"]
tab["sePTSDxCA", "GTP"]<-results.caPTSD[cpg, "logFC"]/results.caPTSD[cpg, "t"]
tab["pPTSDxCA", "GTP"]<-results.caPTSD[cpg, "P.Value"]
tab["FDR_PTSDxCA", "GTP"]<-results.caPTSD[cpg, "adj.P.Val"]
tab["p_1sPTSDxCA", "GTP"]<-ptsdxca.oneSided[cpg,]
pvalue <- as.numeric(results.caPTSD[, "P.Value"])
chisq <- qchisq(1-pvalue,1)
tab["lambdaPTSDxCA", "GTP"]<-median(chisq)/qchisq(0.5,1)
rm(pvalue, chisq)

rm(list=ls()[-match(c("tab", "cpg"), ls())])

# Meta-Analysis

ggd.qqplot = function(pvector, main=NULL, ...) {
  o = -log10(sort(pvector,decreasing=F))
  e = -log10( 1:length(o)/length(o) )
  plot(e,o,pch=19,cex=2, cex.lab=4.5, cex.axis=4, main=main, ...,
       xlab=expression(Expected~~-log[10](italic(p))),
       ylab=expression(Observed~~-log[10](italic(p))),
       xlim=c(0,max(e)), ylim=c(0,max(o)))
  lines(e,e,col="red")
}

load("PGC_EWAS_NRG1_PTSD_mod2_results.Rdata")
res<-pval.combined.marot
colnames(res)<-c("CpG", "p")
pvalue<-as.numeric(res[, "p"])
FDR<-p.adjust(pvalue, method="fdr", n=nrow(res))
res<-cbind(res, FDR)
rownames(res)<-res[, "CpG"]
tab["pPTSD", "Meta"]<-res[cpg, "p"]
tab["FDR_PTSD", "Meta"]<-res[cpg, "FDR"]
pvalue <- as.numeric(pvalue)
chisq <- qchisq(1-pvalue,1)
tab["lambdaPTSD", "Meta"]<-median(chisq)/qchisq(0.5,1)

png("PGC_EWAS_NRG1_PTSD_mod2_QQplot.png", width=1280, height=960, units="px", bg="white")
par(mar=c(10,10,4,2), mgp=c(6,2,0))
ggd.qqplot(pvalue, main = paste("Model 2 PTSD QQ plot"))
dev.off()

rm(list=ls()[-match(c("tab", "cpg", "ggd.qqplot"), ls())])

load("PGC_EWAS_NRG1_CA_mod2_results.Rdata")
res<-pval.combined.marot
colnames(res)<-c("CpG", "p")
pvalue<-as.numeric(res[, "p"])
FDR<-p.adjust(pvalue, method="fdr", n=nrow(res))
res<-cbind(res, FDR)
rownames(res)<-res[, "CpG"]
tab["pCA", "Meta"]<-res[cpg, "p"]
tab["FDR_CA", "Meta"]<-res[cpg, "FDR"]
pvalue <- as.numeric(pvalue)
chisq <- qchisq(1-pvalue,1)
tab["lambdaCA", "Meta"]<-median(chisq)/qchisq(0.5,1)

png("PGC_EWAS_NRG1_CA_mod2_QQplot.png", width=1280, height=960, units="px", bg="white")
par(mar=c(10,10,4,2), mgp=c(6,2,0))
ggd.qqplot(pvalue, main = paste("Model 2 CA QQ plot"))
dev.off()

rm(list=ls()[-match(c("tab", "cpg", "ggd.qqplot"), ls())])

load("PGC_EWAS_NRG1_PTSDxCA_mod2_results.Rdata")
res<-pval.combined.marot
colnames(res)<-c("CpG", "p")
pvalue<-as.numeric(res[, "p"])
FDR<-p.adjust(pvalue, method="fdr", n=nrow(res))
res<-cbind(res, FDR)
rownames(res)<-res[, "CpG"]
tab["pPTSDxCA", "Meta"]<-res[cpg, "p"]
tab["FDR_PTSDxCA", "Meta"]<-res[cpg, "FDR"]
pvalue <- as.numeric(pvalue)
chisq <- qchisq(1-pvalue,1)
tab["lambdaPTSDxCA", "Meta"]<-median(chisq)/qchisq(0.5,1)

png("PGC_EWAS_NRG1_PTSDxCA_mod2_QQplot.png", width=1280, height=960, units="px", bg="white")
par(mar=c(10,10,4,2), mgp=c(6,2,0))
ggd.qqplot(pvalue, main = paste("Model 2 PTSDxCA QQ plot"))
dev.off()

rm(list=ls()[-match(c("tab", "cpg"), ls())])

# Coefficients
betas<-as.numeric(tab["bPTSD", ])
betas<-betas[!is.na(betas)]
stderr<-as.numeric(tab["sePTSD", ])
stderr<-stderr[!is.na(stderr)]
weights<-1/(stderr^2)
tab["bPTSD", "Meta"]<-sum(betas*weights)/sum(weights)
tab["sePTSD", "Meta"]<-1/sum(weights)
rm(betas, stderr, weights)

betas<-as.numeric(tab["bCA", ])
betas<-betas[!is.na(betas)]
stderr<-as.numeric(tab["seCA", ])
stderr<-stderr[!is.na(stderr)]
weights<-1/(stderr^2)
tab["bCA", "Meta"]<-sum(betas*weights)/sum(weights)
tab["seCA", "Meta"]<-1/sum(weights)
rm(betas, stderr, weights)
class(tab)<-"numeric"
tab["N", "Meta"]<-sum(tab["N", ], na.rm=T)

betas<-as.numeric(tab["bPTSDxCA", ])
betas<-betas[!is.na(betas)]
stderr<-as.numeric(tab["sePTSDxCA", ])
stderr<-stderr[!is.na(stderr)]
weights<-1/(stderr^2)
tab["bPTSDxCA", "Meta"]<-sum(betas*weights)/sum(weights)
tab["sePTSDxCA", "Meta"]<-1/sum(weights)
rm(betas, stderr, weights)

write.csv(tab, paste("PGC_EWAS_NRG1_PTSD_mod2_", cpg, ".csv", sep=""))

# PTSD Forest Plot
betas<-tab["bPTSD", c("DNHS", "GTP")]
stderr<-tab["sePTSD", c("DNHS", "GTP")]
lower<-betas-1.96*stderr
upper<-betas+1.96*stderr
betaC<-tab["bPTSD", "Meta"]
stderrC<-tab["sePTSD", "Meta"]
lowerC<-betaC-(1.96*stderrC)
upperC<-betaC+(1.96*stderrC)
beta.table<-c("", "Beta", round(betas,3), NA, round(betaC,3))
subjs<-tab["N", c("DNHS", "GTP")]
Ns<-as.character(unlist(c("", "N", subjs, NA, sum(subjs))))

studies<-c("DNHS", "GTP")
betas<-c(NA, NA, round(betas,3), NA, round(betaC,3))
lower<-c(NA, NA, lower, NA, lowerC)
upper<-c(NA, NA, upper, NA, upperC)
stud<-c(cpg, "Study", studies, NA, "Summary")
summary<-cbind(stud, beta.table, Ns)
summary[which(is.na(summary))]<-""
rownames(summary)<-c(1:nrow(summary))

cochrane<-data.frame(betas, lower, upper)
png(paste("PGC_EWAS_NRG1_PTSD_mod2_", cpg, ".png", sep=""), width=960, height=640,units="px")
forestplot(summary, cochrane,
           boxsize=c(1, 1, 0.2, 0.3), lwd.ci=1.5,
           new_page = TRUE,
           is.summary=c(TRUE,TRUE,rep(FALSE,2),TRUE),
           xlog=FALSE, #lineheight=unit(1,"cm"),
           col=fpColors(box="black",line="black", summary="royalblue"),
           xticks=(c(-0.4, -0.2, 0.1)),
           txt_gp = fpTxtGp( label = list(gpar(fontfamily="", cex=3)),
                             ticks = gpar(fontfamily = "", cex=2)))
dev.off()
rm(list=ls()[-match(c("tab", "cpg"), ls())])

# CA Forest Plot
betas<-tab["bCA", c("DNHS", "GTP")]
stderr<-tab["seCA", c("DNHS", "GTP")]
lower<-betas-1.96*stderr
upper<-betas+1.96*stderr
betaC<-tab["bCA", "Meta"]
stderrC<-tab["seCA", "Meta"]
lowerC<-betaC-(1.96*stderrC)
upperC<-betaC+(1.96*stderrC)
beta.table<-c("", "Beta", round(betas,3), NA, round(betaC,3))
subjs<-tab["N", c("DNHS", "GTP")]
Ns<-as.character(unlist(c("", "N", subjs, NA, sum(subjs))))

studies<-c("DNHS", "GTP")
betas<-c(NA, NA, round(betas,3), NA, round(betaC,3))
lower<-c(NA, NA, lower, NA, lowerC)
upper<-c(NA, NA, upper, NA, upperC)
stud<-c(cpg, "Study", studies, NA, "Summary")
summary<-cbind(stud, beta.table, Ns)
summary[which(is.na(summary))]<-""
rownames(summary)<-c(1:nrow(summary))

cochrane<-data.frame(betas, lower, upper)
png(paste("PGC_EWAS_NRG1_CA_mod2_", cpg, ".png", sep=""), width=960, height=640,units="px")
forestplot(summary, cochrane,
           boxsize=c(1, 1, 0.2, 0.3), lwd.ci=1.5,
           new_page = TRUE,
           is.summary=c(TRUE,TRUE,rep(FALSE,2),TRUE),
           xlog=FALSE, #lineheight=unit(1,"cm"),
           col=fpColors(box="black",line="black", summary="royalblue"),
           xticks=(c(-0.4, -0.2, 0.1)),
           txt_gp = fpTxtGp( label = list(gpar(fontfamily="", cex=3)),
                             ticks = gpar(fontfamily = "", cex=2)))
dev.off()

# PTSDxCA Forest Plot
betas<-tab["bPTSDxCA", c("DNHS", "GTP")]
stderr<-tab["sePTSDxCA", c("DNHS", "GTP")]
lower<-betas-1.96*stderr
upper<-betas+1.96*stderr
betaC<-tab["bPTSDxCA", "Meta"]
stderrC<-tab["sePTSDxCA", "Meta"]
lowerC<-betaC-(1.96*stderrC)
upperC<-betaC+(1.96*stderrC)
beta.table<-c("", "Beta", round(betas,3), NA, round(betaC,3))
subjs<-tab["N", c("DNHS", "GTP")]
Ns<-as.character(unlist(c("", "N", subjs, NA, sum(subjs))))

studies<-c("DNHS", "GTP")
betas<-c(NA, NA, round(betas,3), NA, round(betaC,3))
lower<-c(NA, NA, lower, NA, lowerC)
upper<-c(NA, NA, upper, NA, upperC)
stud<-c(cpg, "Study", studies, NA, "Summary")
summary<-cbind(stud, beta.table, Ns)
summary[which(is.na(summary))]<-""
rownames(summary)<-c(1:nrow(summary))

cochrane<-data.frame(betas, lower, upper)
png(paste("PGC_EWAS_NRG1_PTSDxCA_mod2_", cpg, ".png", sep=""), width=960, height=640,units="px")
forestplot(summary, cochrane,
           boxsize=c(0.8, 0.8, 0.4, 0.6, 0.5), lwd.ci=1.5,
           new_page = TRUE,
           is.summary=c(TRUE,TRUE,rep(FALSE,2),TRUE),
           xlog=FALSE, #lineheight=unit(1,"cm"),
           col=fpColors(box="black",line="black", summary="royalblue"),
           xticks=(c(-0.4, -0.2, 0.1)),
           txt_gp = fpTxtGp( label = list(gpar(fontfamily="", cex=3)),
                             ticks = gpar(fontfamily = "", cex=2)))
dev.off()
```

## Smoking Sensitivity Analysis Code:

### DNHS Smoking Code

```{r, eval=F, echo=T}

rm(list=ls())

########################################################################################
# Step 1: Load Data
########################################################################################

load("/Users/ly2207/Documents/Andrew/R/DNHS/DNHS_Methylation/GenomeStudio/
     DNHS192_bkgd_beta_postComBat_bySample.Rdata")
beta.norm<-reversbeta
rm(reversbeta, missing)

dnhs.p<-read.csv("/Users/ly2207/Documents/Andrew/R/DNHS/DNHS_Methylation/
                 phenotype/DNHS_methylation_pheno_111914_PTSDpm_Trauma_Check.csv", 
                 row.names=1, stringsAsFactors=F)
rownames(dnhs.p)<-dnhs.p$SentrixBarcode_Position

# Subsetting Analytic Sample
cases<-rownames(dnhs.p[dnhs.p$PTSDpm==1,])
controls<-rownames(dnhs.p[dnhs.p$PTSDlife==0, ])
dnhs.p<-dnhs.p[append(cases, controls),]

# Defining Childhood Maltreatment to be two or more types
dnhs.p<-dnhs.p[which(!is.na(dnhs.p$CA_ordinary)),]
dnhs.p$CA_ordinary[dnhs.p$CA_ordinary<2]<-0
dnhs.p$CA_ordinary[dnhs.p$CA_ordinary==2]<-1

p1<-read.csv("/Users/ly2207/Documents/Andrew/R/DNHS/DNHS_Methylation/
             phenotype/DNHS179_meta_pheno_smoking_11.07.16.csv",
             row.names=1, stringsAsFactors=F)
rownames(p1)<-p1$SentrixBarcode_Position

all(rownames(dnhs.p)%in%rownames(p1))
p1<-p1[rownames(dnhs.p), ]
all(rownames(p1)==rownames(dnhs.p))
dnhs.p$CurrSmoke<-p1$CurrSmoke
pheno<-dnhs.p

dim(beta.norm)
beta.norm<-beta.norm[, rownames(pheno)]
dim(beta.norm)
all(rownames(pheno)==colnames(beta.norm))

# Beta values of 0 or 1 will be transformed into M-values of "-Inf". 
range(beta.norm, na.rm=T)

# Changing beta values of 0 to 0.0001
if(min(beta.norm, na.rm=T)==0){
  beta.norm[which(beta.norm==0)]<-0.0001
}

# Changing beta values of 1 to 0.9999
if(max(beta.norm, na.rm=T)==1){
  beta.norm[which(beta.norm==1)]<-0.9999
}

range(beta.norm, na.rm=T)
sum(is.na(beta.norm))

# Convert to Mvalues using log2
beta.norm<-log2(beta.norm/(1-beta.norm)) # log transforming
#sum(beta.norm=="-Inf", na.rm=T) # Should be 0
sum(is.na(beta.norm)) 

# Check that the Subjects in the pheno rows matches order of subjects in Beta columns
all(rownames(pheno)==colnames(beta.norm)) 
sum(is.na(match(rownames(pheno), colnames(beta.norm)))) 
pheno<-pheno[colnames(beta.norm),]
all(rownames(pheno)==colnames(beta.norm)) # this should be true

cpg<-beta.norm["cg23637605", ]
all(names(cpg)==rownames(pheno))
pheno<-cbind(pheno, cpg)
ps<-pheno[!is.na(pheno$CurrSmoke), ]

########################################################################################
# Step 2: Running Models
########################################################################################

fit1<-lm(cpg~age+female+Comp.2+Comp.3+Comp.4+CD8T+CD4T+NK+Bcell+Mono+PTSDpm+
           CA_ordinary, data=pheno)
fit2<-lm(cpg~age+female+Comp.2+Comp.3+Comp.4+CD8T+CD4T+NK+Bcell+Mono+PTSDpm+
           CA_ordinary, data=ps)
fit3<-lm(cpg~age+female+Comp.2+Comp.3+Comp.4+CD8T+CD4T+NK+Bcell+Mono+PTSDpm+
           CA_ordinary+CurrSmoke, data=ps)

tab<-matrix(nrow=6, ncol=3)
colnames(tab)<-c("Main", "Main w Smoke N", "Main + Smoke")
rownames(tab)<-c("bPTSD", "bCA", "bSmoke", "AIC", "N", "ANOVA_pvalue")

cols<-c(1:3)
for(ii in 1:length(cols)){
  res<-get(paste("fit", cols[ii], sep=""))
  sfit<-summary(res)$coef
  
  # PTSD
  beta<-round(sfit["PTSDpm", "Estimate"],3)
  p<-sfit["PTSDpm", "Pr(>|t|)"]
  if(p<0.1){beta<-paste(beta, "^", sep="")}
  if(p<0.05){beta<-paste(beta, "*", sep="")}
  if(p<0.01){beta<-paste(beta, "*", sep="")}
  if(p<0.001){beta<-paste(beta, "*", sep="")}
  tab["bPTSD", cols[ii]]<-beta
  rm(beta,p)
  
  # CA
  beta<-round(sfit["CA_ordinary", "Estimate"],3)
  p<-sfit["CA_ordinary", "Pr(>|t|)"]
  if(p<0.1){beta<-paste(beta, "^", sep="")}
  if(p<0.05){beta<-paste(beta, "*", sep="")}
  if(p<0.01){beta<-paste(beta, "*", sep="")}
  if(p<0.001){beta<-paste(beta, "*", sep="")}
  tab["bCA", cols[ii]]<-beta
  rm(beta,p)
  
  # Smoking
  if(cols[ii]==3){
    beta<-round(sfit["CurrSmoke", "Estimate"],3)
    p<-sfit["CurrSmoke", "Pr(>|t|)"]
    if(p<0.1){beta<-paste(beta, "^", sep="")}
    if(p<0.05){beta<-paste(beta, "*", sep="")}
    if(p<0.01){beta<-paste(beta, "*", sep="")}
    if(p<0.001){beta<-paste(beta, "*", sep="")}
    tab["bSmoke", cols[ii]]<-beta
    rm(beta,p)
  }
  
  # AIC
  tab["AIC", cols[ii]]<-round(AIC(res),3)
  
  # N
  tab["N", cols[ii]]<-round(sum(summary(res)$df[1:2]),3)
}

tab["ANOVA_pvalue", 3]<-round(anova(fit2, fit3)["Pr(>F)"][2,],3)
tab<-gsub("\\^", "", tab)
write.csv(tab, "DNHS_mod1_smoking_results.csv")

### Model 2
rm(list=ls()[-match(c("pheno", "ps"), ls())])

fit1<-lm(cpg~age+female+Comp.2+Comp.3+Comp.4+CD8T+CD4T+NK+Bcell+Mono+PTSDpm
         +CA_ordinary+PTSDpm*CA_ordinary, data=pheno)
fit2<-lm(cpg~age+female+Comp.2+Comp.3+Comp.4+CD8T+CD4T+NK+Bcell+Mono+PTSDpm
         +CA_ordinary+PTSDpm*CA_ordinary, data=ps)
fit3<-lm(cpg~age+female+Comp.2+Comp.3+Comp.4+CD8T+CD4T+NK+Bcell+Mono+PTSDpm
         +CA_ordinary+PTSDpm*CA_ordinary+CurrSmoke, data=ps)

tab<-matrix(nrow=7, ncol=3)
colnames(tab)<-c("Main", "Main w Smoke N", "Main + Smoke")
rownames(tab)<-c("bPTSD", "bCA", "bPTSDxCA", "bSmoke", "AIC", "N", "ANOVA_pvalue")

cols<-c(1:3)
for(ii in 1:length(cols)){
  res<-get(paste("fit", cols[ii], sep=""))
  sfit<-summary(res)$coef
  
  # PTSD
  beta<-round(sfit["PTSDpm", "Estimate"],3)
  p<-sfit["PTSDpm", "Pr(>|t|)"]
  if(p<0.1){beta<-paste(beta, "^", sep="")}
  if(p<0.05){beta<-paste(beta, "*", sep="")}
  if(p<0.01){beta<-paste(beta, "*", sep="")}
  if(p<0.001){beta<-paste(beta, "*", sep="")}
  tab["bPTSD", cols[ii]]<-beta
  rm(beta,p)
  
  # CA
  beta<-round(sfit["CA_ordinary", "Estimate"],3)
  p<-sfit["CA_ordinary", "Pr(>|t|)"]
  if(p<0.1){beta<-paste(beta, "^", sep="")}
  if(p<0.05){beta<-paste(beta, "*", sep="")}
  if(p<0.01){beta<-paste(beta, "*", sep="")}
  if(p<0.001){beta<-paste(beta, "*", sep="")}
  tab["bCA", cols[ii]]<-beta
  rm(beta,p)
  
  # CAxPTSD
  beta<-round(sfit["PTSDpm:CA_ordinary", "Estimate"],3)
  p<-sfit["PTSDpm:CA_ordinary", "Pr(>|t|)"]
  if(p<0.1){beta<-paste(beta, "^", sep="")}
  if(p<0.05){beta<-paste(beta, "*", sep="")}
  if(p<0.01){beta<-paste(beta, "*", sep="")}
  if(p<0.001){beta<-paste(beta, "*", sep="")}
  tab["bPTSDxCA", cols[ii]]<-beta
  rm(beta,p)
  
  # Smoking
  if(cols[ii]==3){
    beta<-round(sfit["CurrSmoke", "Estimate"],3)
    p<-sfit["CurrSmoke", "Pr(>|t|)"]
    if(p<0.1){beta<-paste(beta, "^", sep="")}
    if(p<0.05){beta<-paste(beta, "*", sep="")}
    if(p<0.01){beta<-paste(beta, "*", sep="")}
    if(p<0.001){beta<-paste(beta, "*", sep="")}
    tab["bSmoke", cols[ii]]<-beta
    rm(beta,p)
  }
  
  # AIC
  tab["AIC", cols[ii]]<-round(AIC(res),3)
  
  # N
  tab["N", cols[ii]]<-sum(summary(res)$df[1:2])
}

tab["ANOVA_pvalue", 3]<-round(anova(fit2, fit3)["Pr(>F)"][2,],3)
write.csv(tab, "DNHS_mod2_smoking_results.csv")
```

### GTP Smoking Code

```{r, eval=F, echo=T}

rm(list=ls())

########################################################################################
# Step 1: Load Data
########################################################################################

load("/Users/ly2207/Documents/Andrew/R/GTP/Emory_QC/GTP283_BMIQ_postComBat.Rdata")
beta.norm<-as.matrix(beta)
str(beta.norm)
all(beta.norm==beta, na.rm=T)
rm(beta)

p1<-read.csv("/Users/ly2207/Documents/Andrew/R/GTP/Raw/GTPMetaAnalysisSamples_02.03.17.csv", 
             row.names=1, stringsAsFactors=F)
p2<-read.csv("/Users/ly2207/Documents/Andrew/R/GTP/Raw/DNHS_GTP_Phenotype_for_Analysis_02.26.17.csv", 
             row.names=1, stringsAsFactors=F)
rownames(p2)<-as.character(p2$ID)
int<-intersect(rownames(p1), rownames(p2))
p1<-p1[int, ]
p2<-p2[int, ]
all(rownames(p1)==rownames(p2))

# Adding CA_ordinary
p2$CA_ordinary<-p1$CHILDABphyssexemotandphysemotneg_ctq012_modandsev
p2<-p2[!is.na(p2$CA_ordinary),]
p2$CA_ordinary[p2$CA_ordinary<2]<-0
p2$CA_ordinary[p2$CA_ordinary==2]<-1
gtp.p<-p2[!is.na(p2$PC1), ]
rm(p1, p2)

# Adding CpG site
pheno<-gtp.p
all(rownames(pheno)%in%colnames(beta.norm))
beta.norm<-beta.norm[, rownames(pheno)]

dim(beta.norm)
all(rownames(pheno)==colnames(beta.norm))

# Assuming "beta.norm" is the normalized beta matrix, check the ranges of the beta values. 
# Beta values of 0 or 1 will be transformed into M-values of "-Inf". 
range(beta.norm, na.rm=T)

# Changing beta values of 0 to 0.0001
if(min(beta.norm, na.rm=T)==0){
  beta.norm[which(beta.norm==0)]<-0.0001
}

# Changing beta values of 1 to 0.9999
if(max(beta.norm, na.rm=T)==1){
  beta.norm[which(beta.norm==1)]<-0.9999
}

range(beta.norm, na.rm=T)
sum(is.na(beta.norm))

# Convert to Mvalues using log2
beta.norm<-log2(beta.norm/(1-beta.norm)) # log transforming
#sum(beta.norm=="-Inf", na.rm=T) # Should be 0
#sum(is.na(beta.norm)) # should be same as the number of missing in the beta.norm matrix

# Check that the Subjects in the pheno rows matches order of subjects in Beta columns
all(rownames(pheno)==colnames(beta.norm)) 
sum(is.na(match(rownames(pheno), colnames(beta.norm)))) 
pheno<-pheno[colnames(beta.norm),]
all(rownames(pheno)==colnames(beta.norm)) # this should be true

cpg<-beta.norm["cg23637605", ]
all(names(cpg)==rownames(pheno))
pheno<-cbind(pheno, cpg)
ps<-pheno[!is.na(pheno$SmokerCurr),]

########################################################################################
# Step 2: Running Models
########################################################################################

fit1<-lm(cpg~Age+Gender+Comp.2+Comp.3+Comp.4+CD8T+CD4T+NK+Bcell+Mono+PTSDcurr+CA_ordinary, data=pheno)
fit2<-lm(cpg~Age+Gender+Comp.2+Comp.3+Comp.4+CD8T+CD4T+NK+Bcell+Mono+PTSDcurr+CA_ordinary, data=pheno)
fit3<-lm(cpg~Age+Gender+Comp.2+Comp.3+Comp.4+CD8T+CD4T+NK+Bcell+Mono+PTSDcurr+CA_ordinary+SmokerCurr, data=ps)

tab<-matrix(nrow=6, ncol=3)
colnames(tab)<-c("Main", "Main w Smoke N", "Main + Smoke")
rownames(tab)<-c("bPTSD", "bCA", "bSmoke", "AIC", "N", "ANOVA_pvalue")

cols<-c(1:3)
for(ii in 1:length(cols)){
  res<-get(paste("fit", cols[ii], sep=""))
  sfit<-summary(res)$coef
  
  # PTSD
  beta<-round(sfit["PTSDcurr", "Estimate"],3)
  p<-sfit["PTSDcurr", "Pr(>|t|)"]
  if(p<0.1){beta<-paste(beta, "^", sep="")}
  if(p<0.05){beta<-paste(beta, "*", sep="")}
  if(p<0.01){beta<-paste(beta, "*", sep="")}
  if(p<0.001){beta<-paste(beta, "*", sep="")}
  tab["bPTSD", cols[ii]]<-beta
  rm(beta,p)
  
  # CA
  beta<-round(sfit["CA_ordinary", "Estimate"],3)
  p<-sfit["CA_ordinary", "Pr(>|t|)"]
  if(p<0.1){beta<-paste(beta, "^", sep="")}
  if(p<0.05){beta<-paste(beta, "*", sep="")}
  if(p<0.01){beta<-paste(beta, "*", sep="")}
  if(p<0.001){beta<-paste(beta, "*", sep="")}
  tab["bCA", cols[ii]]<-beta
  rm(beta,p)
  
  # Smoking
  if(cols[ii]==3){
    beta<-round(sfit["SmokerCurr", "Estimate"],3)
    p<-sfit["SmokerCurr", "Pr(>|t|)"]
    if(p<0.1){beta<-paste(beta, "^", sep="")}
    if(p<0.05){beta<-paste(beta, "*", sep="")}
    if(p<0.01){beta<-paste(beta, "*", sep="")}
    if(p<0.001){beta<-paste(beta, "*", sep="")}
    tab["bSmoke", cols[ii]]<-beta
    rm(beta,p)
  }
  
  # AIC
  tab["AIC", cols[ii]]<-round(AIC(res),3)
  
  # N
  tab["N", cols[ii]]<-round(sum(summary(res)$df[1:2]),3)
}

tab["ANOVA_pvalue", 3]<-round(anova(fit2, fit3)["Pr(>F)"][2,],3)
tab<-gsub("\\^", "", tab)

write.csv(tab, "GTP_mod1_smoking_results.csv")

### Model 2
rm(list=ls()[-match(c("pheno", "ps"), ls())])
fit1<-lm(cpg~Age+Gender+Comp.2+Comp.3+Comp.4+CD8T+CD4T+NK+Bcell+Mono+PTSDcurr+CA_ordinary+
           PTSDcurr*CA_ordinary, data=pheno)
fit2<-lm(cpg~Age+Gender+Comp.2+Comp.3+Comp.4+CD8T+CD4T+NK+Bcell+Mono+PTSDcurr+CA_ordinary+
           PTSDcurr*CA_ordinary, data=pheno)
fit3<-lm(cpg~Age+Gender+Comp.2+Comp.3+Comp.4+CD8T+CD4T+NK+Bcell+Mono+PTSDcurr+CA_ordinary+
           PTSDcurr*CA_ordinary+SmokerCurr, data=ps)

tab<-matrix(nrow=7, ncol=3)
colnames(tab)<-c("Main", "Main w Smoke N", "Main + Smoke")
rownames(tab)<-c("bPTSD", "bCA", "bPTSDxCA", "bSmoke", "AIC", "N", "ANOVA_pvalue")

cols<-c(1:3)
for(ii in 1:length(cols)){
  res<-get(paste("fit", cols[ii], sep=""))
  sfit<-summary(res)$coef
  
  # PTSD
  beta<-round(sfit["PTSDcurr", "Estimate"],3)
  p<-sfit["PTSDcurr", "Pr(>|t|)"]
  if(p<0.1){beta<-paste(beta, "^", sep="")}
  if(p<0.05){beta<-paste(beta, "*", sep="")}
  if(p<0.01){beta<-paste(beta, "*", sep="")}
  if(p<0.001){beta<-paste(beta, "*", sep="")}
  tab["bPTSD", cols[ii]]<-beta
  rm(beta,p)
  
  # CA
  beta<-round(sfit["CA_ordinary", "Estimate"],3)
  p<-sfit["CA_ordinary", "Pr(>|t|)"]
  if(p<0.1){beta<-paste(beta, "^", sep="")}
  if(p<0.05){beta<-paste(beta, "*", sep="")}
  if(p<0.01){beta<-paste(beta, "*", sep="")}
  if(p<0.001){beta<-paste(beta, "*", sep="")}
  tab["bCA", cols[ii]]<-beta
  rm(beta,p)
  
  # CAxPTSD
  beta<-round(sfit["PTSDcurr:CA_ordinary", "Estimate"],3)
  p<-sfit["PTSDcurr:CA_ordinary", "Pr(>|t|)"]
  if(p<0.1){beta<-paste(beta, "^", sep="")}
  if(p<0.05){beta<-paste(beta, "*", sep="")}
  if(p<0.01){beta<-paste(beta, "*", sep="")}
  if(p<0.001){beta<-paste(beta, "*", sep="")}
  tab["bPTSDxCA", cols[ii]]<-beta
  rm(beta,p)
  
  # Smoking
  if(cols[ii]==3){
    beta<-round(sfit["SmokerCurr", "Estimate"],3)
    p<-sfit["SmokerCurr", "Pr(>|t|)"]
    if(p<0.1){beta<-paste(beta, "^", sep="")}
    if(p<0.05){beta<-paste(beta, "*", sep="")}
    if(p<0.01){beta<-paste(beta, "*", sep="")}
    if(p<0.001){beta<-paste(beta, "*", sep="")}
    tab["bSmoke", cols[ii]]<-beta
    rm(beta,p)
  }
  
  # AIC
  tab["AIC", cols[ii]]<-round(AIC(res),3)
  
  # N
  tab["N", cols[ii]]<-sum(summary(res)$df[1:2])
}

tab["ANOVA_pvalue", 3]<-round(anova(fit2, fit3)["Pr(>F)"][2,],3)
tab<-gsub("\\^", "", tab)
write.csv(tab, "GTP_mod2_smoking_results.csv")
```

\newpage

# References
